diff -dur irda-utils-0.9.16/irattach/dongle_attach.c irda-utils-0.9.16-p/irattach/dongle_attach.c
--- irda-utils-0.9.16/irattach/dongle_attach.c	2002-06-23 20:09:31.000000000 +0000
+++ irda-utils-0.9.16-p/irattach/dongle_attach.c	2004-10-27 19:06:50.718390704 +0000
@@ -38,6 +38,19 @@
 #include <signal.h>
 #include <string.h>
 
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <asm/byteorder.h>
 
 #include <net/if.h>
diff -dur irda-utils-0.9.16/irdaping/irdaping.c irda-utils-0.9.16-p/irdaping/irdaping.c
--- irda-utils-0.9.16/irdaping/irdaping.c	2003-07-08 01:33:39.000000000 +0000
+++ irda-utils-0.9.16-p/irdaping/irdaping.c	2004-10-27 19:07:13.140981952 +0000
@@ -43,6 +43,19 @@
  * the only ones that knows anything about IrDA and byte ordering.
  */
 
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <asm/byteorder.h>	/* __cpu_to_le32 and co. */
 
 #include <linux/types.h>	/* For __u8 and co. */
diff -dur irda-utils-0.9.16/tekram/irkbd.c irda-utils-0.9.16-p/tekram/irkbd.c
--- irda-utils-0.9.16/tekram/irkbd.c	2000-11-06 21:13:52.000000000 +0000
+++ irda-utils-0.9.16-p/tekram/irkbd.c	2004-10-27 19:05:01.285027104 +0000
@@ -45,6 +45,18 @@
 
 #include <irda.h>
 
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <asm/byteorder.h>
 
 #include "irkbd.h"
