--- ./src/seltrans.h.orig	2007-11-16 17:09:32.000000000 +0000
+++ ./src/seltrans.h	2007-11-16 17:10:07.000000000 +0000
@@ -138,8 +138,8 @@
     NR::Point _origin; ///< position of origin for transforms
     NR::Point _point; ///< original position of the knot being used for the current transform
     Inkscape::MessageContext _message_context;
-    SigC::Connection _sel_changed_connection;
-    SigC::Connection _sel_modified_connection;
+    sigc::connection _sel_changed_connection;
+    sigc::connection _sel_modified_connection;
 };
 
 }
