--- ./src/anvil/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/anvil/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/bounce/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/bounce/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -20,7 +20,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/cleanup/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/cleanup/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -24,7 +24,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/discard/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/discard/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/dns/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/dns/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -19,7 +19,7 @@
 all: $(LIB)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/error/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/error/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/flush/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/flush/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/fsstone/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/fsstone/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -15,7 +15,7 @@
 all:	$(PROG)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 fsstone: fsstone.o $(LIBS)
 	$(CC) $(CFLAGS) -o $@ fsstone.o $(LIBS) $(SYSLIBS)
--- ./src/global/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/global/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -98,7 +98,7 @@
 all: $(LIB)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/lmtp/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/lmtp/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -20,7 +20,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/local/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/local/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -22,7 +22,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/master/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/master/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -25,7 +25,7 @@
 all:	$(PROG) $(LIB)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 $(PROG): $(OBJS) $(LIBS)
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
--- ./src/oqmgr/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/oqmgr/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -20,7 +20,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/pickup/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/pickup/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/pipe/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/pipe/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/postalias/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postalias/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 update: ../../bin/$(PROG)
 
--- ./src/postcat/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postcat/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/postconf/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postconf/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -26,7 +26,7 @@
 	 ./$(PROG) -d) |egrep -v '^(myhostname|mydomain|mynetworks) ' >$@
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/postdrop/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postdrop/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/postfix/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postfix/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -17,7 +17,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/postkick/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postkick/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/postlock/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postlock/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/postlog/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postlog/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -17,7 +17,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/postmap/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postmap/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 update: ../../bin/$(PROG)
 
--- ./src/postqueue/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postqueue/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/postsuper/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/postsuper/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/proxymap/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/proxymap/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/qmgr/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/qmgr/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -22,7 +22,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/tls/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/tls/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -25,7 +25,7 @@
 all: $(LIB)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/qmqpd/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/qmqpd/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/scache/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/scache/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/sendmail/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/sendmail/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/showq/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/showq/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/smtp/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/smtp/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -21,7 +21,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/smtpd/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/smtpd/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -22,7 +22,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/smtpstone/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/smtpstone/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -15,7 +15,7 @@
 all:	$(PROG)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 smtp-sink: smtp-sink.o $(LIBS)
 	$(CC) $(CFLAGS) -o $@ smtp-sink.o $(LIBS) $(SYSLIBS)
--- ./src/spawn/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/spawn/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/trivial-rewrite/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/trivial-rewrite/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -21,7 +21,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/util/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/util/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -110,7 +110,7 @@
 all: $(LIB)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/verify/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/verify/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./src/virtual/Makefile.in.orig	2005-03-11 18:17:48.000000000 +0100
+++ ./src/virtual/Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS) $(SYSLIBS)
 
 Makefile: Makefile.in
-	(echo "# DO NOT EDIT"; tail +2 ../../conf/makedefs.out; cat $?) >$@
+	(echo "# DO NOT EDIT"; tail -n +2 ../../conf/makedefs.out; cat $?) >$@
 
 test:	$(TESTPROG)
 
--- ./Makefile.in.orig	2005-02-02 21:44:15.000000000 +0100
+++ ./Makefile.in	2005-03-11 18:20:02.000000000 +0100
@@ -20,7 +20,7 @@
 	 (set -e; echo "[$$i]"; cd $$i; rm -f Makefile; \
 	 $(MAKE) -f Makefile.in Makefile MAKELEVEL=) || exit 1; \
 	done;
-	rm -f Makefile; (tail +2 conf/makedefs.out; cat Makefile.in) >Makefile
+	rm -f Makefile; (tail -n +2 conf/makedefs.out; cat Makefile.in) >Makefile
 
 update printfck tests:
 	set -e; for i in $(DIRS); do \
