diff -ruN ffmpeg-0.4.9-pre1/Makefile ffmpeg-0.4.9-pre1-new/Makefile
--- ffmpeg-0.4.9-pre1/Makefile	2004-07-08 13:02:04.000000000 +0200
+++ ffmpeg-0.4.9-pre1-new/Makefile	2004-07-26 12:55:08.261050798 +0200
@@ -14,6 +14,9 @@
 LDFLAGS+=-p
 endif
 
+# ReneR: well the maintainer could also test his ... without imlib2 support
+EXTRALIBS+=-L/usr/X11/lib -lX11
+
 MANPAGE=$(SRC_PATH)/doc/ffmpeg.1
 PROG=ffmpeg$(EXESUF)
 PROGTEST=output_example$(EXESUF)
