--- ./misc/Makefile.am.orig	2004-12-01 22:15:59.768577368 +0100
+++ ./misc/Makefile.am	2004-12-01 22:16:18.036800176 +0100
@@ -27,7 +27,7 @@
 	chmod a+x xine-check
 
 xine-bugreport: xine-check
-	test ! -f xine-bugreport && $(LN_S) xine-check xine-bugreport
+	if test ! -f xine-bugreport ; then $(LN_S) xine-check xine-bugreport ; fi
 
 
 debug:
--- ./misc/Makefile.in.orig	2004-12-01 22:17:13.376387280 +0100
+++ ./misc/Makefile.in	2004-12-01 22:17:31.632611912 +0100
@@ -623,7 +623,7 @@
 	chmod a+x xine-check
 
 xine-bugreport: xine-check
-	test ! -f xine-bugreport && $(LN_S) xine-check xine-bugreport
+	if test ! -f xine-bugreport ; then $(LN_S) xine-check xine-bugreport ; fi
 
 debug:
 	@list='$(SUBDIRS)'; for subdir in $$list; do \
