--- ./crm/admin/Makefile.in.orig	2006-04-03 17:46:21.000000000 +0200
+++ ./crm/admin/Makefile.in	2006-04-03 17:47:50.000000000 +0200
@@ -466,8 +466,7 @@
 noinst_HEADERS = 
 crmadmin_SOURCES = crmadmin.c
 crmadmin_CFLAGS = -DHA_VARLIBDIR='"@HA_VARLIBDIR@"'
-crmadmin_LDADD = $(COMMONLIBS) \
-			$(top_builddir)/crm/pengine/libpengine.la
+crmadmin_LDADD = $(top_builddir)/crm/pengine/libpengine.la $(COMMONLIBS)
 
 crm_uuid_SOURCES = crm_uuid.c
 crm_uuid_CFLAGS = -DHA_VARLIBDIR='"@HA_VARLIBDIR@"'
@@ -490,8 +489,7 @@
 
 crm_verify_SOURCES = crm_verify.c
 crm_verify_CFLAGS = -DHA_VARLIBDIR='"@HA_VARLIBDIR@"'
-crm_verify_LDADD = $(COMMONLIBS) \
-			$(top_builddir)/crm/pengine/libpengine.la
+crm_verify_LDADD = $(top_builddir)/crm/pengine/libpengine.la $(COMMONLIBS)
 
 crm_master_SOURCES = crm_attribute.c
 crm_master_CFLAGS = -DHA_VARLIBDIR='"@HA_VARLIBDIR@"'
@@ -504,8 +502,7 @@
 crm_attribute_LDADD = $(COMMONLIBS)
 crm_resource_SOURCES = crm_resource.c
 crm_resource_CFLAGS = -DHA_VARLIBDIR='"@HA_VARLIBDIR@"'
-crm_resource_LDADD = $(COMMONLIBS) \
-			$(top_builddir)/crm/pengine/libpengine.la
+crm_resource_LDADD = $(top_builddir)/crm/pengine/libpengine.la $(COMMONLIBS)
 
 iso8601_SOURCES = test.iso8601.c
 iso8601_CFLAGS = -DHA_VARLIBDIR='"@HA_VARLIBDIR@"'
