diff -dur samba-3.0.20-orig/source/configure samba-3.0.20/source/configure
--- source/configure	2005-08-19 21:09:01.000000000 +0200
+++ source/configure	2005-09-29 10:30:23.000000000 +0200
@@ -4041,7 +4041,7 @@
 
 
 if test "$ac_cv_prog_gnu_ld" = "yes"; then
-	ac_cv_gnu_ld_version=`$LD -v 2>/dev/null | head -1`
+	ac_cv_gnu_ld_version=`$LD -v 2>/dev/null | head -n1`
 	echo "$as_me:$LINENO: checking GNU ld release date" >&5
 echo $ECHO_N "checking GNU ld release date... $ECHO_C" >&6
 		ac_cv_gnu_ld_date=`echo $ac_cv_gnu_ld_version | sed -n 's,^.*\([2-9][0-9][0-9][0-9]\)[-]*\([01][0-9]\)[-]*\([0-3][0-9]\).*$,\1\2\3,p'`
diff -dur samba-3.0.20-orig/source/configure.in samba-3.0.20/source/configure.in
--- source/configure.in	2005-08-19 19:16:29.000000000 +0200
+++ source/configure.in	2005-09-29 10:30:09.000000000 +0200
@@ -350,7 +350,7 @@
 dnl --allow-shlib-undefined flag defined.  This causes a stackload of
 dnl warnings when building modules.
 if test "$ac_cv_prog_gnu_ld" = "yes"; then
-	ac_cv_gnu_ld_version=`$LD -v 2>/dev/null | head -1`
+	ac_cv_gnu_ld_version=`$LD -v 2>/dev/null | head -n1`
 	AC_MSG_CHECKING(GNU ld release date)
 	changequote(,)dnl
 	ac_cv_gnu_ld_date=`echo $ac_cv_gnu_ld_version | sed -n 's,^.*\([2-9][0-9][0-9][0-9]\)[-]*\([01][0-9]\)[-]*\([0-3][0-9]\).*$,\1\2\3,p'`
