diff -pruN man-pages-1.67_orig/Makefile man-pages-1.67/Makefile
--- man-pages-1.67_orig/Makefile	2004-05-21 01:35:18.000000000 +0200
+++ man-pages-1.67/Makefile	2004-10-21 11:41:08.226029864 +0200
@@ -17,10 +17,8 @@ allbz: bz2 all
 screen:
 	-mkdir not_installed
 	for i in man?/*; do \
-		if [ $(MANDIR)/"$$i" -nt "$$i" ]; then \
-			cmp -s $(MANDIR)/"$$i" "$$i" > /dev/null 2>&1; \
-			if [ "$$?" != 0 ]; then mv "$$i" not_installed; fi; \
-		fi; \
+		grep -q "$$i" ` ls /var/adm/flists/* | grep -v man-pages `; \
+		if [ "$$?" == 0 ]; then mv "$$i" not_installed; fi; \
 	done
 
 remove:
