--- xc/programs/Xserver/hw/tinyx/linux/linux.c	2004-06-27 13:01:19.000000000 +0200
+++ xc-p/programs/Xserver/hw/tinyx/linux/linux.c	2004-11-11 22:20:32.022231840 +0100
@@ -125,10 +125,10 @@
 	vtno = kdVirtualTerminal;
     else
     {
-	if ((fd = open("/dev/tty0",O_WRONLY,0)) < 0) 
+	if ((fd = open("/dev/vc/0",O_WRONLY,0)) < 0) 
 	{
 	    FatalError(
-		       "LinuxInit: Cannot open /dev/tty0 (%s)\n",
+		       "LinuxInit: Cannot open /dev/vc/0 (%s)\n",
 		       strerror(errno));
 	}
 	if ((ioctl(fd, VT_OPENQRY, &vtno) < 0) ||
@@ -139,7 +139,7 @@
 	close(fd);
     }
 
-    sprintf(vtname,"/dev/tty%d",vtno); /* /dev/tty1-64 */
+    sprintf(vtname,"/dev/vc/%d",vtno); /* /dev/vc/1-64 */
 
     if ((LinuxConsoleFd = open(vtname, O_RDWR|O_NDELAY, 0)) < 0)
     {
@@ -156,7 +156,7 @@
      *
      * Why is this needed?
      */
-    LinuxCheckChown ("/dev/tty0");
+    LinuxCheckChown ("/dev/vc/0");
     /*
      * Linux doesn't switch to an active vt after the last close of a vt,
      * so we do this ourselves by remembering which is active now.
@@ -483,7 +483,7 @@
 	activeVT = -1;
     }
     close(LinuxConsoleFd);                /* make the vt-manager happy */
-    fd = open ("/dev/tty0", O_RDWR|O_NDELAY, 0);
+    fd = open ("/dev/vc/0", O_RDWR|O_NDELAY, 0);
     if (fd >= 0)
     {
 	ioctl (fd, VT_GETSTATE, &vts);
