--- ./bindings/c/tag_c.cpp.orig	2008-06-10 19:26:34.000000000 +0200
+++ ./bindings/c/tag_c.cpp	2008-06-10 19:26:50.000000000 +0200
@@ -21,6 +21,8 @@
 
 #include "tag_c.h"
 
+#include <string.h>
+
 #include <fileref.h>
 #include <tfile.h>
 #include <vorbisfile.h>
--- ./taglib/toolkit/tbytevector.cpp.orig	2008-06-10 19:24:06.000000000 +0200
+++ ./taglib/toolkit/tbytevector.cpp	2008-06-10 19:24:14.000000000 +0200
@@ -20,6 +20,7 @@
  ***************************************************************************/
 
 #include <iostream>
+#include <string.h>
 
 #include <tstring.h>
 #include <tdebug.h>
--- ./taglib/toolkit/tfile.cpp.orig	2008-06-10 19:24:28.000000000 +0200
+++ ./taglib/toolkit/tfile.cpp	2008-06-10 19:24:49.000000000 +0200
@@ -24,6 +24,8 @@
 #include "tdebug.h"
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <sys/stat.h>
 #include <unistd.h>
 
--- ./taglib/toolkit/tstring.cpp.orig	2008-06-10 19:23:21.000000000 +0200
+++ ./taglib/toolkit/tstring.cpp	2008-06-10 19:23:46.000000000 +0200
@@ -24,6 +24,7 @@
 #include "tdebug.h"
 
 #include <iostream>
+#include <string.h>
 
 namespace TagLib {
 
