--- ./python/Makefile.am.orig	2007-09-20 01:03:25.000000000 +0200
+++ ./python/Makefile.am	2007-09-20 01:03:47.000000000 +0200
@@ -40,7 +40,7 @@
 	export PYTHONPATH=$$builddir/pyqt:$$PYTHONPATH ; \
 	cd $(srcdir)/pyqt ; \
 	export QTDIR=${qt_dir} ; \
-	echo "yes" | @PYTHON@ configure.py -c -b @bindir@ -d ${site_packages} -v ${sip_dir} -t "$$builddir/pyqt"
+	echo "yes" | @PYTHON@ configure.py -j 100 -c -b @bindir@ -d ${site_packages} -v ${sip_dir} -t "$$builddir/pyqt"
 	echo "" > prepqt
 
 prepkde: prepqt
@@ -54,7 +54,7 @@
 	export PYTHONPATH=$$builddir/pykde:$$PYTHONPATH ; \
 	cd $(srcdir)/pykde ; \
 	export QTDIR=${qt_dir} ; \
-	@PYTHON@ configure.py -c -k @prefix@ -d ${site_packages} -v ${sip_dir} -t "$$builddir/pykde"
+	@PYTHON@ configure.py -j 100 -c -k @prefix@ -d ${site_packages} -v ${sip_dir} -t "$$builddir/pykde"
 	echo "" > prepkde
 
 install:
--- ./python/Makefile.in.orig	2007-09-20 01:02:26.000000000 +0200
+++ ./python/Makefile.in	2007-09-20 01:03:21.000000000 +0200
@@ -593,7 +593,7 @@
 	export PYTHONPATH=$$builddir/pyqt:$$PYTHONPATH ; \
 	cd $(srcdir)/pyqt ; \
 	export QTDIR=${qt_dir} ; \
-	echo "yes" | @PYTHON@ configure.py -c -b @bindir@ -d ${site_packages} -v ${sip_dir} -t "$$builddir/pyqt"
+	echo "yes" | @PYTHON@ configure.py -j 100 -c -b @bindir@ -d ${site_packages} -v ${sip_dir} -t "$$builddir/pyqt"
 	echo "" > prepqt
 
 prepkde: prepqt
@@ -607,7 +607,7 @@
 	export PYTHONPATH=$$builddir/pykde:$$PYTHONPATH ; \
 	cd $(srcdir)/pykde ; \
 	export QTDIR=${qt_dir} ; \
-	@PYTHON@ configure.py -c -k @prefix@ -d ${site_packages} -v ${sip_dir} -t "$$builddir/pykde"
+	@PYTHON@ configure.py -j 100 -c -k @prefix@ -d ${site_packages} -v ${sip_dir} -t "$$builddir/pykde"
 	echo "" > prepkde
 
 install:
