--- valgrind-2.2.0/coregrind/vg_unsafe.h	2004-08-29 21:02:28.000000000 +0000
+++ valgrind-2.2.0-p/coregrind/vg_unsafe.h	2004-10-24 08:47:50.959307152 +0000
@@ -47,7 +47,14 @@
 #include <sys/socket.h>   /* for struct msghdr */
 #include <linux/sockios.h>/* for SIOCOUTQ */
 #include <sys/un.h>       /* for sockaddr_un */
-#include <net/if.h>       /* for struct ifreq et al */
+
+/* for linux-2.6.9, from linux/types.h */
+#  include <asm/types.h>
+#  ifndef pgoff_t
+#  define pgoff_t unsigned long
+#  endif
+
+/* #include <net/if.h> */ /* for struct ifreq et al */
 #include <net/if_arp.h>   /* for struct arpreq */
 #include <net/route.h>    /* for struct rtentry */
 #ifdef HAVE_LINUX_COMPILER_H
