diff -dur valgrind-2.2.0/coregrind/vg_unsafe.h valgrind-2.2.0-p/coregrind/vg_unsafe.h
--- valgrind-2.2.0/coregrind/vg_unsafe.h	2004-11-02 22:03:17.000000000 +0100
+++ valgrind-2.2.0-p/coregrind/vg_unsafe.h	2004-11-02 20:49:06.000000000 +0100
@@ -47,7 +47,17 @@
 #include <sys/socket.h>   /* for struct msghdr */
 #include <linux/sockios.h>/* for SIOCOUTQ */
 #include <sys/un.h>       /* for sockaddr_un */
+
+/* for linux-2.6.9, from linux/types.h */
+#  include <asm/types.h>
+#  ifndef pgoff_t
+#  define pgoff_t unsigned long
+#  endif
+
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,9)
 #include <net/if.h>       /* for struct ifreq et al */
+#endif
 #include <net/if_arp.h>   /* for struct arpreq */
 #include <net/route.h>    /* for struct rtentry */
 #ifdef HAVE_LINUX_COMPILER_H
