diff -ruN courier-imap-3.0.8/Makefile.am courier-imap-3.0.8-new/Makefile.am
--- courier-imap-3.0.8/Makefile.am	2004-08-29 21:03:28.000000000 +0200
+++ courier-imap-3.0.8-new/Makefile.am	2004-10-13 21:41:38.837692912 +0200
@@ -21,7 +21,7 @@
 databindir=$(datadir)
 databin_SCRIPTS=mkimapdcert mkpop3dcert vchkpw2userdb makeuserdb pw2userdb
 
-binPROGRAMS=imapd pop3d maildirmake maildiracl deliverquota maildirkw
+binPROGRAMS=imapd pop3d maildiracl deliverquota maildirkw
 sbinPROGRAMS=authenumerate imaplogin pop3login userdbpw courierlogger
 libexecPROGRAMS=makedatprog couriertcpd
 
@@ -31,7 +31,7 @@
 
 EXTRA_PROGRAMS=
 
-man_MANS=makeuserdb.8 userdb.8 userdbpw.8 couriertcpd.1 courierlogger.1 deliverquota.8 maildirmake.1 maildiracl.1 imapd.8 mkimapdcert.8 mkpop3dcert.8 maildirkw.1
+man_MANS=makeuserdb.8 userdb.8 userdbpw.8 couriertcpd.1 courierlogger.1 deliverquota.8 maildiracl.1 imapd.8 mkimapdcert.8 mkpop3dcert.8 maildirkw.1
 EXTRA_DIST=00README.NOW.OR.SUFFER README README.cygwin automake.fix \
 	COPYING.GPL NEWS.html $(TEMPLFILES)
 
@@ -89,9 +89,6 @@
 vchkpw2userdb: userdb/vchkpw2userdb
 	cp userdb/vchkpw2userdb vchkpw2userdb
 
-maildirmake$(EXEEXT): maildir/maildirmake$(EXEEXT)
-	cp maildir/maildirmake$(EXEEXT) .
-
 maildiracl$(EXEEXT): maildir/maildiracl$(EXEEXT)
 	cp maildir/maildiracl$(EXEEXT) .
 
@@ -128,9 +125,6 @@
 deliverquota.8: maildir/deliverquota.8
 	cp maildir/deliverquota.8 .
 
-maildirmake.1: maildir/maildirmake.1
-	cp maildir/maildirmake.1 .
-
 maildiracl.1: maildir/maildiracl.1
 	cp maildir/maildiracl.1 .
 
diff -ruN courier-imap-3.0.8/Makefile.in courier-imap-3.0.8-new/Makefile.in
--- courier-imap-3.0.8/Makefile.in	2004-09-18 20:49:13.000000000 +0200
+++ courier-imap-3.0.8-new/Makefile.in	2004-10-13 21:42:16.521964032 +0200
@@ -203,10 +203,10 @@
 CLEANFILES = $(databin_SCRIPTS) $(man_MANS) $(sysconf_DATA) $(sbin_SCRIPTS)
 databindir = $(datadir)
 databin_SCRIPTS = mkimapdcert mkpop3dcert vchkpw2userdb makeuserdb pw2userdb
-binPROGRAMS = imapd pop3d maildirmake maildiracl deliverquota maildirkw
+binPROGRAMS = imapd pop3d maildiracl deliverquota maildirkw
 sbinPROGRAMS = authenumerate imaplogin pop3login userdbpw courierlogger
 libexecPROGRAMS = makedatprog couriertcpd
-man_MANS = makeuserdb.8 userdb.8 userdbpw.8 couriertcpd.1 courierlogger.1 deliverquota.8 maildirmake.1 maildiracl.1 imapd.8 mkimapdcert.8 mkpop3dcert.8 maildirkw.1
+man_MANS = makeuserdb.8 userdb.8 userdbpw.8 couriertcpd.1 courierlogger.1 deliverquota.8 maildiracl.1 imapd.8 mkimapdcert.8 mkpop3dcert.8 maildirkw.1
 
 # Added by sysconftoolize: $Id: Makefile.am,v 1.96 2004/08/29 19:03:28 mrsam Exp $
 
@@ -948,9 +948,6 @@
 vchkpw2userdb: userdb/vchkpw2userdb
 	cp userdb/vchkpw2userdb vchkpw2userdb
 
-maildirmake$(EXEEXT): maildir/maildirmake$(EXEEXT)
-	cp maildir/maildirmake$(EXEEXT) .
-
 maildiracl$(EXEEXT): maildir/maildiracl$(EXEEXT)
 	cp maildir/maildiracl$(EXEEXT) .
 
@@ -987,9 +984,6 @@
 deliverquota.8: maildir/deliverquota.8
 	cp maildir/deliverquota.8 .
 
-maildirmake.1: maildir/maildirmake.1
-	cp maildir/maildirmake.1 .
-
 maildiracl.1: maildir/maildiracl.1
 	cp maildir/maildiracl.1 .
 
