diff -dur linux-2.6.17-orig/scripts/kconfig/conf.c linux-2.6.17/scripts/kconfig/conf.c
--- linux-2.6.17-orig/scripts/kconfig/conf.c	2006-06-18 03:49:35.000000000 +0200
+++ linux-2.6.17/scripts/kconfig/conf.c	2006-07-27 22:13:38.000000000 +0200
@@ -21,6 +21,7 @@
 	ask_all,
 	ask_new,
 	ask_silent,
+	ask_no2mod,
 	set_default,
 	set_yes,
 	set_mod,
@@ -92,6 +93,15 @@
 			return;
 		}
 		break;
+       case ask_no2mod:
+               if (type == S_TRISTATE &&
+                   sym_get_tristate_value(sym) == no) {
+                       fprintf(stderr, "Setting %s to 'm'.\n", sym->name);
+                       line[0] = 'm';
+                       line[1] = '\n';
+                       line[2] = 0;
+                       break;
+               }
 	case ask_new:
 	case ask_silent:
 		if (sym_has_value(sym)) {
@@ -354,6 +364,7 @@
 				break;
 			}
 			check_stdin();
+		case ask_no2mod:
 		case ask_all:
 			fflush(stdout);
 			fgets(line, 128, stdin);
@@ -523,6 +534,9 @@
 		case 'm':
 			input_mode = set_mod;
 			break;
+		case 'M':
+			input_mode = ask_no2mod;
+			break;
 		case 'y':
 			input_mode = set_yes;
 			break;
@@ -565,6 +579,7 @@
 		}
 	case ask_all:
 	case ask_new:
+	case ask_no2mod:
 		conf_read(NULL);
 		break;
 	case set_no:
diff -dur linux-2.6.17-orig/scripts/kconfig/Makefile linux-2.6.17/scripts/kconfig/Makefile
--- linux-2.6.17-orig/scripts/kconfig/Makefile	2006-06-18 03:49:35.000000000 +0200
+++ linux-2.6.17/scripts/kconfig/Makefile	2006-07-27 22:15:10.000000000 +0200
@@ -42,7 +42,7 @@
 	$(Q)rm -f arch/um/Kconfig_arch
 	$(Q)rm -f scripts/kconfig/linux_*.pot scripts/kconfig/config.pot
 
-PHONY += randconfig allyesconfig allnoconfig allmodconfig defconfig
+PHONY += randconfig allyesconfig allnoconfig allmodconfig no2modconfig defconfig
 
 randconfig: $(obj)/conf
 	$< -r $(Kconfig)
@@ -56,6 +56,9 @@
 allmodconfig: $(obj)/conf
 	$< -m $(Kconfig)
 
+no2modconfig: scripts/kconfig/conf
+	$< -M $(Kconfig)
+
 defconfig: $(obj)/conf
 ifeq ($(KBUILD_DEFCONFIG),)
 	$< -d $(Kconfig)
@@ -77,6 +80,7 @@
 	@echo  '  randconfig	  - New config with random answer to all options'
 	@echo  '  defconfig	  - New config with default answer to all options'
 	@echo  '  allmodconfig	  - New config selecting modules when possible'
+	@echo  '  no2modconfig    - New config selecting modules for disabled options'
 	@echo  '  allyesconfig	  - New config where all options are accepted with yes'
 	@echo  '  allnoconfig	  - New config where all options are answered with no'
 
