--- ./include/asm-sparc/byteorder.h.orig	2004-11-21 14:52:27.000000000 +0100
+++ ./include/asm-sparc/byteorder.h	2004-12-02 22:37:37.459072968 +0100
@@ -4,10 +4,8 @@
 
 #include <asm/types.h>
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__) || defined(__KERNEL__)
-#  define __BYTEORDER_HAS_U64__
-#  define __SWAB_64_THRU_32__
-#endif
+#define __BYTEORDER_HAS_U64__
+#define __SWAB_64_THRU_32__
 
 #include <linux/byteorder/big_endian.h>
 
