--- ./coda-src/venus/fso.h.orig	2004-09-11 20:23:20.808096008 +0200
+++ ./coda-src/venus/fso.h	2004-09-11 20:24:28.690776272 +0200
@@ -539,7 +539,7 @@
     void DetachHdbBindings();
     void DetachHdbBinding(binding *, int =0);
     int PredetermineFetchState(int, int);
-    void SetFetchAllowed(enum HoardFetchState new_state)
+    void SetFetchAllowed(HoardFetchState new_state)
         { FetchAllowed = new_state; }
     int IsFetchAllowed()
         { if (FetchAllowed == HF_Fetch) 
@@ -547,7 +547,7 @@
           else 
               return 0; 
         }
-    void SetAskingAllowed(enum HoardAskState new_state)
+    void SetAskingAllowed(HoardAskState new_state)
         { AskingAllowed = new_state; }
     int IsAskingAllowed()
         { if (AskingAllowed == HA_Ask)
--- ./coda-src/venus/comm.cc.orig	2004-09-11 20:24:39.570122360 +0200
+++ ./coda-src/venus/comm.cc	2004-09-11 20:24:47.975844496 +0200
@@ -173,7 +173,7 @@
     sei.AckPoint = sftp_ackpoint;
     sei.PacketSize = sftp_packetsize;
     sei.EnforceQuota = 1;
-    sei.Port.Tag = (enum PortTag)0;
+    sei.Port.Tag = (PortTag)0;
     SFTP_Activate(&sei);
 
     /* RPC2 initialization. */
