diff -dur shadow-4.0.5/src/Makefile.in shadow-4.0.5-p/src/Makefile.in
--- shadow-4.0.5/src/Makefile.in	2004-10-26 03:34:09.000000000 +0200
+++ shadow-4.0.5-p/src/Makefile.in	2004-11-07 04:56:47.042845024 +0100
@@ -352,8 +352,10 @@
 # id and groups are from gnu, sulogin from sysvinit
 suidbins = su
 suidubins = chage chfn chsh expiry gpasswd newgrp passwd
+# Add $(LIBPAM) to $(LDADD) because libmisc and libshadow are using some
+# functions provided by the PAM library.
 LDADD = $(top_builddir)/libmisc/libmisc.a \
-	$(top_builddir)/lib/libshadow.la
+	$(top_builddir)/lib/libshadow.la $(LIBPAM)
 
 chpasswd_LDADD = $(LDADD) $(LIBPAM)
 chage_LDADD = $(LDADD) $(LIBPAM)
