--- ./compat/include/pathnames.h~	2005-10-19 05:53:56.000000000 +0200
+++ ./compat/include/pathnames.h	2005-10-19 05:55:24.000000000 +0200
@@ -47,11 +47,11 @@
 
 #ifndef _PATH_DEFTAPE
 #ifdef __linux__
-#define	_PATH_DEFTAPE	"/dev/st0"
+#define	_PATH_DEFTAPE	"/dev/tape/tape0"
 #endif
 #ifdef sunos
 #define	_PATH_DEFTAPE	"/dev/rmt/0"
 #endif
 #endif
 
-#define	_PATH_RMT	"/etc/rmt"		/* path on remote host */
+#define	_PATH_RMT	"/sbin/rmt"		/* path on remote host */
