diff -dur ppp-2.4.2b3/pppd/plugins/rp-pppoe/plugin.c ppp-2.4.2b3-p/pppd/plugins/rp-pppoe/plugin.c
--- ppp-2.4.2b3/pppd/plugins/rp-pppoe/plugin.c	2003-04-07 00:01:46.000000000 +0000
+++ ppp-2.4.2b3-p/pppd/plugins/rp-pppoe/plugin.c	2004-10-27 00:33:18.449399456 +0000
@@ -50,6 +50,19 @@
 #include <net/if_arp.h>
 #include "ppp_defs.h"
 #include "if_ppp.h"
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include "if_pppox.h"
 
 #define _PATH_ETHOPT         _ROOT_PATH "/etc/ppp/options."
