--- ./src/ui_about.ui.orig	2005-10-31 18:07:00.000000000 +0100
+++ ./src/ui_about.ui	2005-10-31 18:07:06.000000000 +0100
@@ -334,7 +334,6 @@
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
     <includehint>fancylabel.h</includehint>
-    <includehint>urllabel.h</includehint>
     <includehint>psitextview.h</includehint>
     <includehint>psitextview.h</includehint>
     <includehint>psitextview.h</includehint>
--- ./src/options/opt_appearance-ui.ui.orig	2005-10-31 18:11:43.000000000 +0100
+++ ./src/options/opt_appearance-ui.ui	2005-10-31 18:11:53.000000000 +0100
@@ -458,9 +458,5 @@
 </tabstops>
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
-    <includehint>fontlabel.h</includehint>
-    <includehint>fontlabel.h</includehint>
-    <includehint>fontlabel.h</includehint>
-    <includehint>fontlabel.h</includehint>
 </includehints>
 </UI>
