diff -ruN kdissert-1.0.5/src/kdissert/canvasview/DCanvasLink.cpp kdissert-1.0.5-new/src/kdissert/canvasview/DCanvasLink.cpp
--- kdissert-1.0.5/src/kdissert/canvasview/DCanvasLink.cpp	2005-10-22 02:43:14.000000000 +0400
+++ kdissert-1.0.5-new/src/kdissert/canvasview/DCanvasLink.cpp	2005-11-15 02:43:35.000000000 +0300
@@ -13,6 +13,7 @@
  ***************************************************************************/
 
 #include <math.h>
+#include <stdlib.h>
 #include <qpainter.h>
 #include <qcolor.h>
 #include <kdebug.h>
