--- ./Makefile.orig	2004-08-06 22:29:09.403635888 +0200
+++ ./Makefile	2004-08-06 22:29:58.195218440 +0200
@@ -5,6 +5,8 @@
 CFLAGS = -g -O2 -Wall -fomit-frame-pointer
 LDFLAGS =
 
+PREFIX = /usr/local
+
 SRC = iostat.c
 BIN = iostat
 
@@ -14,9 +16,9 @@
 	$(CC) $(CFLAGS) $(SRC) -o $@ $(LDFLAGS)
 
 install: $(BIN)
-	install -d /usr/local/bin /usr/local/man/man8
-	install -o root -g root -m 755 -s iostat /usr/local/bin
-	install -o root -g root -m 644 iostat.8 /usr/local/man/man8
+	install -d $(PREFIX)/bin $(PREFIX)/man/man8
+	install -o root -g root -m 755 -s iostat $(PREFIX)/bin
+	install -o root -g root -m 644 iostat.8 $(PREFIX)/man/man8
 
 clean:
 	rm -f $(BIN) core
