--- ./noteedit/timesigDia.ui.orig	2005-10-31 18:39:26.000000000 +0100
+++ ./noteedit/timesigDia.ui	2005-10-31 18:39:30.000000000 +0100
@@ -357,7 +357,5 @@
 </slots>
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
 </includehints>
 </UI>
--- ./noteedit/staffProps.ui.orig	2005-10-31 18:39:09.000000000 +0100
+++ ./noteedit/staffProps.ui	2005-10-31 18:39:15.000000000 +0100
@@ -543,13 +543,5 @@
 </slots>
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
 </includehints>
 </UI>
