diff -Naur xemacs-21.5.17.orig/lib-src/movemail.c xemacs-21.5.17/lib-src/movemail.c
--- xemacs-21.5.17.orig/lib-src/movemail.c	2002-03-13 08:51:59.000000000 +0000
+++ xemacs-21.5.17/lib-src/movemail.c	2005-02-09 09:37:39.000000000 +0000
@@ -801,7 +801,7 @@
           mbx_delimit_begin (mbf);
 	  if (pop_retr (server, i, mbx_write, mbf) != POP_RETRIEVED)
 	    {
-	      error (Errmsg, NULL, NULL);
+	      error ("%s", Errmsg, NULL, NULL);
 	      close (mbfi);
 	      return (1);
 	    }
