*** ./pngconf.h_125	2002-10-03 07:32:27.000000000 -0400
--- ./pngconf.h	2004-05-06 09:41:33.000000000 -0400
***************
*** 1280,1285 ****
--- 1285,1291 ----
  #  define CVT_PTR(ptr) (png_far_to_near(png_ptr,ptr,CHECK))
  #  define CVT_PTR_NOCHECK(ptr) (png_far_to_near(png_ptr,ptr,NOCHECK))
  #  define png_strcpy _fstrcpy
+ #  define png_strncpy _fstrncpy    /* Added to v 1.2.6 */
  #  define png_strlen _fstrlen
  #  define png_memcmp _fmemcmp      /* SJT: added */
  #  define png_memcpy _fmemcpy
***************
*** 1288,1293 ****
--- 1294,1300 ----
  #  define CVT_PTR(ptr)         (ptr)
  #  define CVT_PTR_NOCHECK(ptr) (ptr)
  #  define png_strcpy strcpy
+ #  define png_strncpy strncpy   /* Added to v 1.2.6 */
  #  define png_strlen strlen
  #  define png_memcmp memcmp     /* SJT: added */
  #  define png_memcpy memcpy
*** ./pngerror.c_125	2002-10-03 07:32:27.000000000 -0400
--- ./pngerror.c	2004-05-06 09:41:28.000000000 -0400
***************
*** 137,143 ****
     {
        buffer[iout++] = ':';
        buffer[iout++] = ' ';
!       png_memcpy(buffer+iout, error_message, 64);
        buffer[iout+63] = 0;
     }
  }
--- 137,143 ----
     {
        buffer[iout++] = ':';
        buffer[iout++] = ' ';
!       png_strncpy(buffer+iout, error_message, 63);
        buffer[iout+63] = 0;
     }
  }
