--- ./noatun-plugins/dub/dub/random.h.orig	2008-06-21 08:03:16.000000000 +0200
+++ ./noatun-plugins/dub/dub/random.h	2008-06-21 08:03:38.000000000 +0200
@@ -15,6 +15,7 @@
 #define Random_Interface
 
 #include <cstdlib>
+#include <time.h>
 
 // wrapper for random functions
 class Random {
--- ./noatun-plugins/noatunmadness/madness.cpp.orig	2008-06-21 08:09:47.000000000 +0200
+++ ./noatun-plugins/noatunmadness/madness.cpp	2008-06-21 08:09:53.000000000 +0200
@@ -14,7 +14,7 @@
 #include <kwinmodule.h>
 #include <kiconloader.h>
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 
 #include <X11/Xlib.h>
 
--- ./noatun-plugins/tippercanoe/sdlwrap.cpp.orig	2008-06-21 08:14:04.000000000 +0200
+++ ./noatun-plugins/tippercanoe/sdlwrap.cpp	2008-06-21 08:14:09.000000000 +0200
@@ -23,7 +23,7 @@
 #include <time.h>
 #include <string.h>
 #include "syna.h"
-#include <iostream.h>
+#include <iostream>
 #include <SDL_syswm.h>
 #include "SDL.h"
 
