--- ./plug-ins/common/Makefile.in.orig	2004-12-19 19:21:17.134739008 +0100
+++ ./plug-ins/common/Makefile.in	2004-12-19 19:23:42.682612368 +0100
@@ -1374,6 +1374,7 @@
 	$(libgimpcolor)		\
 	$(libgimpbase)		\
 	$(GLIB_LIBS)		\
+	$(GTK_LIBS)		\
 	$(RT_LIBS)		\
 	$(INTLLIBS)
 
@@ -1524,6 +1526,7 @@
 	$(libgimpcolor)		\
 	$(libgimpbase)		\
 	$(GLIB_LIBS)		\
+	$(GTK_LIBS)		\
 	$(RT_LIBS)		\
 	$(INTLLIBS)
 
