--- ./daemon/gdm-xdmcp-manager.c~	2007-11-08 20:42:21.000000000 +0000
+++ ./daemon/gdm-xdmcp-manager.c	2007-11-08 20:43:00.000000000 +0000
@@ -63,11 +63,11 @@
  * On Sun, we need to define allow_severity and deny_severity to link
  * against libwrap.
  */
-#ifdef __sun
+// #ifdef __sun
 #include <syslog.h>
 int allow_severity = LOG_INFO;
 int deny_severity = LOG_WARNING;
-#endif
+// #endif
 
 #define GDM_XDMCP_MANAGER_GET_PRIVATE(o) (G_TYPE_INSTANCE_GET_PRIVATE ((o), GDM_TYPE_XDMCP_MANAGER, GdmXdmcpManagerPrivate))
 
