diff -pruN udev-084-orig/etc/udev/debian/devfs.rules udev-084/etc/udev/debian/devfs.rules
--- udev-084-orig/etc/udev/debian/devfs.rules	2006-02-14 16:11:51.000000000 +0100
+++ udev-084/etc/udev/debian/devfs.rules	2006-02-14 16:13:23.000000000 +0100
@@ -96,7 +96,7 @@ KERNEL=="pktcdvd",		NAME="pktcdvd/contro
 
 KERNEL=="cpu[0-9]*",		NAME="cpu/%n/cpuid"
 KERNEL=="msr[0-9]*",		NAME="cpu/%n/msr"
-KERNEL=="microcode",		NAME="cpu/microcode"
+KERNEL=="microcode",		NAME="cpu/microcode", SYMLINK="misc/microcode"
 
 KERNEL=="umad*",		NAME="infiniband/%k"
 KERNEL=="issm*",		NAME="infiniband/%k"
