--- ./etc/udev/debian/devfs.rules.orig	2006-01-13 00:23:21.195562168 +0100
+++ ./etc/udev/debian/devfs.rules	2006-01-13 00:24:18.676823688 +0100
@@ -18,12 +18,14 @@
 	ENV{ID_MODEL}=="IOMEGA_ZIP*",	NAME="%k", OPTIONS+="all_partitions"
 
 # IDE devices
-BUS=="ide", KERNEL=="hd[a-z]*", PROGRAM="/lib/udev/ide-devfs.sh %k %b %n", \
-					NAME="%c{1}", SYMLINK+="%k %c{2}"
+BUS=="ide", KERNEL=="hd[a-z]*", PROGRAM="/etc/udev/scripts/ide-devfs.sh %k %b %n", \
+ 				NAME="%c{1}", SYMLINK+="%k %c{2}", GROUP="%c{3}"
 
 # SCSI devices
-BUS=="scsi", PROGRAM="/lib/udev/scsi-devfs.sh %k %b %n", \
-					NAME="%c{1}", SYMLINK+="%k %c{2}"
+BUS=="scsi", KERNEL=="sd*", PROGRAM="/etc/udev/scripts/scsi-devfs.sh sd %b %n", NAME="%c{1}", SYMLINK="%c{2} %k %c{3}"
+BUS=="scsi", KERNEL=="sr*", PROGRAM="/etc/udev/scripts/scsi-devfs.sh sr %b %n", NAME="%c{1}", SYMLINK="%c{2} %k %c{3}"
+BUS=="scsi", KERNEL=="st*", PROGRAM="/etc/udev/scripts/scsi-devfs.sh st %b %n", NAME="%c{1}", SYMLINK="%c{2} %k %c{3}"
+BUS=="scsi", KERNEL=="sg*", PROGRAM="/etc/udev/scripts/scsi-devfs.sh sg %b %n", NAME="%c{1}", SYMLINK="%c{2} %k %c{3}"
 
 # RAID controllers
 KERNEL=="cciss!*|ida!*|rd!*", PROGRAM="/lib/udev/raid-devfs.sh %k", \
