diff -pruN udev-068-orig/extras/scsi-devfs.sh udev-068/extras/scsi-devfs.sh
--- udev-068-orig/extras/scsi-devfs.sh	2005-10-11 17:33:26.000000000 +0200
+++ udev-068/extras/scsi-devfs.sh	2005-10-12 10:17:44.000000000 +0200
@@ -83,6 +83,16 @@ l_pci="domain$pci_domain/bus$pci_bus/slo
 l_phy="bus/pci/$l_pci/scsi/$l_com"
 s_phy="$1/pci/$pci_addr/$s_com"
 
-echo $l_phy $s_phy $l_log $s_log
+for x in /dev/discs/disc* ; do
+       if readlink $x/disc | grep -q "$l_pci" ; then
+               x=`echo $x | cut -f3 -dc` # gives the number in disc0
+               break
+       fi
+       unset x
+done
 
+if [ -z "${x}" ] ; then
+	x="`ls /dev/discs/ | grep -c .`"
+fi
 
+echo $l_phy $s_phy $l_log $s_log discs/disc${x}/${lpart}
