diff -pruN udev-077/extras/ide-devfs.sh udev-077-patched/extras/ide-devfs.sh
--- udev-077/extras/ide-devfs.sh	2005-12-04 03:01:48.000000000 +0100
+++ udev-077-patched/extras/ide-devfs.sh	2006-01-16 08:54:51.000000000 +0100
@@ -20,7 +20,7 @@ get_dev_number() {
 
 	for x in /proc/ide/*/media; do
 		if [ -e "${x}" ]; then
-			lMEDIA=`cat ${x}`
+			read lMEDIA < ${x}
 			if [ "${lMEDIA}" = "$2" ]; then
 				num=$((${num} + 1))
 			fi
@@ -34,7 +34,7 @@ get_dev_number() {
 }
 
 if [ -z "$3" ] && [ -f /proc/ide/${1}/media ]; then
-	MEDIA=`cat /proc/ide/${1}/media`
+	read MEDIA < /proc/ide/${1}/media
 	if [ "${MEDIA}" = "cdrom" ]; then
 		echo ide/host${HOST}/bus${BUS}/target${TARGET}/lun0/cd cdroms/cdrom`get_dev_number $1 cdrom` ${MEDIA}
 	elif [ "${MEDIA}" = "floppy" ]; then
diff -pruN udev-077/extras/raid-devfs.sh udev-077-patched/extras/raid-devfs.sh
--- udev-077/extras/raid-devfs.sh	2005-12-04 03:01:48.000000000 +0100
+++ udev-077-patched/extras/raid-devfs.sh	2006-01-16 09:01:47.000000000 +0100
@@ -9,17 +9,29 @@
 # KERNEL=="rd!*",    PROGRAM="/etc/udev/scripts/raid-devfs.sh %k", NAME="%c{1}", SYMLINK+="%k"
 
 get_dev_number_cciss() {
-  grep '^cciss/' /proc/driver/cciss/* | cat -n | \
+  num=1
+  grep '^cciss/' /proc/driver/cciss/* | while read line ; do
+	echo "$num\t$line"
+	num=$(($num+1))
+  done | \
     sed -n "/cciss\/$1:/s/cciss.*//p"
 }
 
 get_dev_number_ida() {
-  grep '^ida/' /proc/driver/cpqarray/* | cat -n | \
+  num=1
+  grep '^ida/' /proc/driver/cpqarray/* | while read line ; do
+	echo "$num\t$line"
+	num=$(($num+1))
+  done | \
     sed -n "/ida\/$1:/s/ida.*//p"
 }
 
 get_dev_number_rd() {
-  grep '^ */dev/rd/' /proc/rd/*/initial_status | cat -n | \
+  num=1
+  grep '^ */dev/rd/' /proc/rd/*/initial_status | while read line ; do
+	echo "$num\t$line"
+	num=$(($num+1))
+  done | \
     sed -n "/rd\/$1:/s/\/dev\/rd.*//p"
 }
 
