diff -dur disktype-6/cdaccess.c disktype-6-p/cdaccess.c
--- disktype-6/cdaccess.c	2003-06-01 12:45:47.000000000 +0000
+++ disktype-6-p/cdaccess.c	2004-10-24 05:14:37.645186520 +0000
@@ -32,6 +32,19 @@
 
 #ifdef USE_IOCTL_LINUX
 #include <sys/ioctl.h>
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <linux/cdrom.h>
 #define DO_CDACCESS 1
 #endif  /* USE_IOCTL_LINUX */
