--- ./configure.orig	2007-05-15 22:36:30.000000000 +0200
+++ ./configure	2007-05-15 22:37:42.000000000 +0200
@@ -4591,7 +4591,7 @@
  	for dir in $SEARCHPATH /usr/local/freetype /usr/local /opt/freetype /usr/pkg /usr /usr/include/freetype1; do
  		ac_cv_ft_lib=no
  		ac_cv_ft_include=no
-		shlib=`echo $dir/lib/libttf.$SHLIBEXT* | head -1 | sed -e 's/.*\///'`
+		shlib=`echo $dir/lib/libttf.$SHLIBEXT* | head -n 1 | sed -e 's/.*\///'`
 		if test x$shlib = x; then
 			shlib=no
 		fi
@@ -4896,8 +4896,8 @@
 		ac_cv_vf_libname=VFlib
 		ac_cv_vf_libdir=no
 		ac_cv_vf_hdrdir=no
-		shlib1=`echo $dir/lib/libVFlib.so* | head -1 | sed -e 's/.*\///'`
-		shlib2=`echo $dir/lib/libVFlib2.so* | head -1 | sed -e 's/.*\///'`
+		shlib1=`echo $dir/lib/libVFlib.so* | head -n 1 | sed -e 's/.*\///'`
+		shlib2=`echo $dir/lib/libVFlib2.so* | head -n 1 | sed -e 's/.*\///'`
 		if test x$shlib1 = x; then
 			$shlib1=no
 		fi
