--- ./src/Makefile.orig	2004-12-29 00:31:53.034057536 +0100
+++ ./src/Makefile	2004-12-29 00:33:07.428747824 +0100
@@ -26,7 +26,7 @@
 ifeq ($(DISTRO),)
 BIN	+= mountpoint
 SBIN	+= sulogin bootlogd
-USRBIN	+= utmpdump wall
+USRBIN	+= wall
 MAN1	+= mountpoint.1 wall.1
 MAN8	+= sulogin.8 bootlogd.8
 endif
@@ -71,9 +71,6 @@
 mountpoint:	mountpoint.o
 		$(CC) $(LDFLAGS) -o $@ mountpoint.o
 
-utmpdump:	utmpdump.o
-		$(CC) $(LDFLAGS) -o $@ utmpdump.o
-
 runlevel:	runlevel.o
 		$(CC) $(LDFLAGS) -o $@ runlevel.o
 
