diff -dur dbus-glib-0.71/tools/Makefile.in build/crystal-distro-k6III-TRUNK-x86-k6-3-32-crystal-expert/ROCK/src.dbus-glib.1154591626.24357.3315736450/dbus-glib-0.71/tools/Makefile.in
--- dbus-glib-0.71/tools/Makefile.in	2006-07-24 20:41:57.000000000 +0200
+++ build/crystal-distro-k6III-TRUNK-x86-k6-3-32-crystal-expert/ROCK/src.dbus-glib.1154591626.24357.3315736450/dbus-glib-0.71/tools/Makefile.in	2006-08-03 10:20:14.000000000 +0200
@@ -487,7 +487,7 @@
 	$(top_builddir)/dbus/dbus-binding-tool --mode=glib-client --output=dbus-glib-bindings.h dbus-bus-introspect.xml
 
 dbus-bus-introspect.xml:
-	DBUS_TOP_BUILDDIR=$(top_builddir) dbus-send --system --print-reply=literal --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect > dbus-bus-introspect.xml.tmp && mv dbus-bus-introspect.xml.tmp dbus-bus-introspect.xml
+	DBUS_TOP_BUILDDIR=$(top_builddir) ./run-with-tmp-session-bus.sh dbus-send --session --print-reply=literal --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect > dbus-bus-introspect.xml.tmp && mv dbus-bus-introspect.xml.tmp dbus-bus-introspect.xml
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -dur dbus-glib-0.71/tools/session.conf build/crystal-distro-k6III-TRUNK-x86-k6-3-32-crystal-expert/ROCK/src.dbus-glib.1154591626.24357.3315736450/dbus-glib-0.71/tools/session.conf
--- dbus-glib-0.71/tools/session.conf	2006-08-03 10:20:39.000000000 +0200
+++ build/crystal-distro-k6III-TRUNK-x86-k6-3-32-crystal-expert/ROCK/src.dbus-glib.1154591626.24357.3315736450/dbus-glib-0.71/tools/session.conf	2006-07-28 21:00:04.000000000 +0200
@@ -0,0 +1,30 @@
+<!-- This configuration file controls the per-user-login-session message bus.
+     Add a session-local.conf and edit that rather than changing this 
+     file directly. -->
+
+<!DOCTYPE busconfig PUBLIC "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
+ "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
+<busconfig>
+  <!-- Our well-known bus type, don't change this -->
+  <type>session</type>
+
+  <listen>unix:tmpdir=/tmp</listen>
+
+  <servicedir>/usr/share/dbus-1/services</servicedir>
+
+  <policy context="default">
+    <!-- Allow everything to be sent -->
+    <allow send_destination="*"/>
+    <!-- Allow everything to be received -->
+    <allow eavesdrop="true"/>
+    <!-- Allow anyone to own anything -->
+    <allow own="*"/>
+  </policy>
+
+  <!-- This is included last so local configuration can override what's 
+       in this standard file -->
+  <include ignore_missing="yes">session-local.conf</include>
+
+  <include if_selinux_enabled="yes" selinux_root_relative="yes">contexts/dbus_contexts</include>
+
+</busconfig>
--- dbus-glib-0.71/tools/Makefile.am	2006-07-17 19:00:39.000000000 +0200
+++ build/crystal-distro-k6III-TRUNK-x86-k6-3-32-crystal-expert/ROCK/src.dbus-glib.1154591626.24357.3315736450/dbus-glib-0.71/tools/Makefile.am	2006-08-03 10:24:54.000000000 +0200
@@ -9,7 +9,7 @@
 BUILT_SOURCES = dbus-glib-bindings.h dbus-bus-introspect.xml
 
 dbus-bus-introspect.xml:
-	DBUS_TOP_BUILDDIR=$(top_builddir) dbus-send --system --print-reply=literal --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect > dbus-bus-introspect.xml.tmp && mv dbus-bus-introspect.xml.tmp dbus-bus-introspect.xml
+	DBUS_TOP_BUILDDIR=$(top_builddir) ./run-with-tmp-session-bus.sh dbus-send --session --print-reply=literal --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect > dbus-bus-introspect.xml.tmp && mv dbus-bus-introspect.xml.tmp dbus-bus-introspect.xml
 
 EXTRA_DIST = run-with-tmp-session-bus.sh
 
