--- tvtime-0.9.12/src/Makefile.am.orig	Thu Dec  4 04:11:19 2003
+++ tvtime-0.9.12/src/Makefile.am	Thu Dec  4 04:11:34 2003
@@ -70,7 +70,7 @@
 	performance.c vbidata.h vbidata.c credits.c credits.h console.h \
 	console.c speedtools.h vbiscreen.h vbiscreen.c fifo.h fifo.c \
 	commands.h commands.c videofilter.h videofilter.c station.h station.c \
-	bands.h vsync.h vsync.c utils.h utils.c diffcomp.h diffcomp.c \
+	bands.h utils.h utils.c diffcomp.h diffcomp.c \
 	rvrreader.h rvrreader.c ree.h ree.c pulldown.h pulldown.c \
 	hashtable.h hashtable.c cpuinfo.h cpuinfo.c videodev.h videodev2.h \
 	menu.c menu.h outputfilter.h outputfilter.c dfbmatroxtv.c xmltv.h \
--- tvtime-0.9.12/src/Makefile.in.orig	Thu Dec  4 04:21:55 2003
+++ tvtime-0.9.12/src/Makefile.in	Thu Dec  4 05:11:29 2003
@@ -195,7 +195,7 @@
 	performance.c vbidata.h vbidata.c credits.c credits.h console.h \
 	console.c speedtools.h vbiscreen.h vbiscreen.c fifo.h fifo.c \
 	commands.h commands.c videofilter.h videofilter.c station.h station.c \
-	bands.h vsync.h vsync.c utils.h utils.c diffcomp.h diffcomp.c \
+	bands.h utils.h utils.c diffcomp.h diffcomp.c \
 	rvrreader.h rvrreader.c ree.h ree.c pulldown.h pulldown.c \
 	hashtable.h hashtable.c cpuinfo.h cpuinfo.c videodev.h videodev2.h \
 	menu.c menu.h outputfilter.h outputfilter.c dfbmatroxtv.c xmltv.h \
@@ -300,7 +300,7 @@
 	timingtest-credits.$(OBJEXT) timingtest-console.$(OBJEXT) \
 	timingtest-vbiscreen.$(OBJEXT) timingtest-fifo.$(OBJEXT) \
 	timingtest-commands.$(OBJEXT) timingtest-videofilter.$(OBJEXT) \
-	timingtest-station.$(OBJEXT) timingtest-vsync.$(OBJEXT) \
+	timingtest-station.$(OBJEXT) \
 	timingtest-utils.$(OBJEXT) timingtest-diffcomp.$(OBJEXT) \
 	timingtest-rvrreader.$(OBJEXT) timingtest-ree.$(OBJEXT) \
 	timingtest-pulldown.$(OBJEXT) timingtest-hashtable.$(OBJEXT) \
@@ -323,7 +323,7 @@
 	tvtime-console.$(OBJEXT) tvtime-vbiscreen.$(OBJEXT) \
 	tvtime-fifo.$(OBJEXT) tvtime-commands.$(OBJEXT) \
 	tvtime-videofilter.$(OBJEXT) tvtime-station.$(OBJEXT) \
-	tvtime-vsync.$(OBJEXT) tvtime-utils.$(OBJEXT) \
+	tvtime-utils.$(OBJEXT) \
 	tvtime-diffcomp.$(OBJEXT) tvtime-rvrreader.$(OBJEXT) \
 	tvtime-ree.$(OBJEXT) tvtime-pulldown.$(OBJEXT) \
 	tvtime-hashtable.$(OBJEXT) tvtime-cpuinfo.$(OBJEXT) \
@@ -416,7 +416,6 @@
 @AMDEP_TRUE@	./$(DEPDIR)/timingtest-videofilter.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/timingtest-videoinput.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/timingtest-videotools.Po \
-@AMDEP_TRUE@	./$(DEPDIR)/timingtest-vsync.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/timingtest-xmltv.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/tvtime-commands.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/tvtime-console.Po \
@@ -458,7 +457,6 @@
 @AMDEP_TRUE@	./$(DEPDIR)/tvtime-videofilter.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/tvtime-videoinput.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/tvtime-videotools.Po \
-@AMDEP_TRUE@	./$(DEPDIR)/tvtime-vsync.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/tvtime-xcommon.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/tvtime-xfullscreen.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/tvtime-xmltv.Po \
@@ -569,7 +567,6 @@
 timingtest-commands.$(OBJEXT): commands.c
 timingtest-videofilter.$(OBJEXT): videofilter.c
 timingtest-station.$(OBJEXT): station.c
-timingtest-vsync.$(OBJEXT): vsync.c
 timingtest-utils.$(OBJEXT): utils.c
 timingtest-diffcomp.$(OBJEXT): diffcomp.c
 timingtest-rvrreader.$(OBJEXT): rvrreader.c
@@ -610,7 +607,6 @@
 tvtime-commands.$(OBJEXT): commands.c
 tvtime-videofilter.$(OBJEXT): videofilter.c
 tvtime-station.$(OBJEXT): station.c
-tvtime-vsync.$(OBJEXT): vsync.c
 tvtime-utils.$(OBJEXT): utils.c
 tvtime-diffcomp.$(OBJEXT): diffcomp.c
 tvtime-rvrreader.$(OBJEXT): rvrreader.c
@@ -707,7 +703,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/timingtest-videofilter.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/timingtest-videoinput.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/timingtest-videotools.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/timingtest-vsync.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/timingtest-xmltv.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tvtime-commands.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tvtime-console.Po@am__quote@
@@ -749,7 +744,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tvtime-videofilter.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tvtime-videoinput.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tvtime-videotools.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tvtime-vsync.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tvtime-xcommon.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tvtime-xfullscreen.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tvtime-xmltv.Po@am__quote@
@@ -1345,24 +1339,6 @@
 @AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(timingtest_CFLAGS) $(CFLAGS) -c -o timingtest-station.lo `test -f 'station.c' || echo '$(srcdir)/'`station.c
 
-timingtest-vsync.o: vsync.c
-@AMDEP_TRUE@	source='vsync.c' object='timingtest-vsync.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/timingtest-vsync.Po' tmpdepfile='$(DEPDIR)/timingtest-vsync.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(timingtest_CFLAGS) $(CFLAGS) -c -o timingtest-vsync.o `test -f 'vsync.c' || echo '$(srcdir)/'`vsync.c
-
-timingtest-vsync.obj: vsync.c
-@AMDEP_TRUE@	source='vsync.c' object='timingtest-vsync.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/timingtest-vsync.Po' tmpdepfile='$(DEPDIR)/timingtest-vsync.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(timingtest_CFLAGS) $(CFLAGS) -c -o timingtest-vsync.obj `cygpath -w vsync.c`
-
-timingtest-vsync.lo: vsync.c
-@AMDEP_TRUE@	source='vsync.c' object='timingtest-vsync.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/timingtest-vsync.Plo' tmpdepfile='$(DEPDIR)/timingtest-vsync.TPlo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(timingtest_CFLAGS) $(CFLAGS) -c -o timingtest-vsync.lo `test -f 'vsync.c' || echo '$(srcdir)/'`vsync.c
-
 timingtest-utils.o: utils.c
 @AMDEP_TRUE@	source='utils.c' object='timingtest-utils.o' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@	depfile='$(DEPDIR)/timingtest-utils.Po' tmpdepfile='$(DEPDIR)/timingtest-utils.TPo' @AMDEPBACKSLASH@
@@ -2029,24 +2005,6 @@
 @AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(tvtime_CFLAGS) $(CFLAGS) -c -o tvtime-station.lo `test -f 'station.c' || echo '$(srcdir)/'`station.c
 
-tvtime-vsync.o: vsync.c
-@AMDEP_TRUE@	source='vsync.c' object='tvtime-vsync.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/tvtime-vsync.Po' tmpdepfile='$(DEPDIR)/tvtime-vsync.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(tvtime_CFLAGS) $(CFLAGS) -c -o tvtime-vsync.o `test -f 'vsync.c' || echo '$(srcdir)/'`vsync.c
-
-tvtime-vsync.obj: vsync.c
-@AMDEP_TRUE@	source='vsync.c' object='tvtime-vsync.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/tvtime-vsync.Po' tmpdepfile='$(DEPDIR)/tvtime-vsync.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(tvtime_CFLAGS) $(CFLAGS) -c -o tvtime-vsync.obj `cygpath -w vsync.c`
-
-tvtime-vsync.lo: vsync.c
-@AMDEP_TRUE@	source='vsync.c' object='tvtime-vsync.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/tvtime-vsync.Plo' tmpdepfile='$(DEPDIR)/tvtime-vsync.TPlo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(tvtime_CFLAGS) $(CFLAGS) -c -o tvtime-vsync.lo `test -f 'vsync.c' || echo '$(srcdir)/'`vsync.c
-
 tvtime-utils.o: utils.c
 @AMDEP_TRUE@	source='utils.c' object='tvtime-utils.o' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@	depfile='$(DEPDIR)/tvtime-utils.Po' tmpdepfile='$(DEPDIR)/tvtime-utils.TPo' @AMDEPBACKSLASH@
