--- ./linux/Makefile.in	2005-03-23 17:45:24.000000000 +0100
+++ ./linux/Makefile.in	2005-06-18 13:19:23.000000000 +0200
@@ -65,7 +65,7 @@
 	$(INSTALL) -m 444 ../include/AL/alut.h $(DESTINC)/AL
 	$(INSTALL) -m 444 ../include/AL/altypes.h $(DESTINC)/AL
 	$(INSTALL) -m 444 ../include/AL/alctypes.h $(DESTINC)/AL
-	$(INSTALL) -m 444 ../include/AL/aluttypes.h $(DESTINC)/AL
+#	$(INSTALL) -m 444 ../include/AL/aluttypes.h $(DESTINC)/AL
 	$(INSTALL) -m 444 include/AL/alext.h $(DESTINC)/AL
 	$(INSTALL) -m 444 include/AL/alexttypes.h $(DESTINC)/AL
 
