--- ./src/core/export_project_dialog.cpp.orig	2006-03-16 12:14:02.000000000 +0100
+++ ./src/core/export_project_dialog.cpp	2006-03-16 12:14:30.000000000 +0100
@@ -46,6 +46,7 @@
 
 #endif
 
+#include <stdlib.h>
 
 #include "export_project_dialog.h"
 #include "song_editor.h"
diff -dur lmms-0.1.4/src/core/piano_roll.cpp build/generic-distro-TRUNK-x86-athlon-xp-32-generic-expert/ROCK/src.lmms.1143553855.6854.3820088609/lmms-0.1.4/src/core/piano_roll.cpp
--- lmms-0.1.4/src/core/piano_roll.cpp	2006-01-22 13:35:35.000000000 +0100
+++ build/generic-distro-TRUNK-x86-athlon-xp-32-generic-expert/ROCK/src.lmms.1143553855.6854.3820088609/lmms-0.1.4/src/core/piano_roll.cpp	2006-03-29 22:48:41.000000000 +0200
@@ -23,7 +23,6 @@
  *
  */
 
-
 #include "qt3support.h"
 
 #ifdef QT4
@@ -70,6 +69,10 @@
 #include "text_float.h"
 #include "combobox.h"
 
+#ifdef LMMS_DEBUG
+#include <assert.h>
+#endif
+
 
 extern tones whiteKeys[];	// defined in piano_widget.cpp
 
diff -dur lmms-0.1.4/src/lib/ladspa_manager.cpp build/generic-distro-TRUNK-x86-athlon-xp-32-generic-expert/ROCK/src.lmms.1143553855.6854.3820088609/lmms-0.1.4/src/lib/ladspa_manager.cpp
--- lmms-0.1.4/src/lib/ladspa_manager.cpp	2005-12-19 13:17:23.000000000 +0100
+++ build/generic-distro-TRUNK-x86-athlon-xp-32-generic-expert/ROCK/src.lmms.1143553855.6854.3820088609/lmms-0.1.4/src/lib/ladspa_manager.cpp	2006-03-29 22:52:16.000000000 +0200
@@ -46,6 +46,7 @@
 #endif
 
 
+#include <stdlib.h>
 #include <math.h>
 
 
diff -dur lmms-0.1.4/src/midi/midi_file.cpp build/generic-distro-TRUNK-x86-athlon-xp-32-generic-expert/ROCK/src.lmms.1143553855.6854.3820088609/lmms-0.1.4/src/midi/midi_file.cpp
--- lmms-0.1.4/src/midi/midi_file.cpp	2005-12-15 11:04:24.000000000 +0100
+++ build/generic-distro-TRUNK-x86-athlon-xp-32-generic-expert/ROCK/src.lmms.1143553855.6854.3820088609/lmms-0.1.4/src/midi/midi_file.cpp	2006-03-29 22:53:20.000000000 +0200
@@ -44,6 +44,10 @@
 #include <qapplication.h>
 #include <qprogressdialog.h>
 
+#ifdef LMMS_DEBUG
+#include <assert.h>
+#endif
+
 #define ungetChar ungetch
 #define seek at 
 #define pos at
