--- ./arts/builder/structure.cpp.orig	2008-06-13 11:24:39.000000000 +0200
+++ ./arts/builder/structure.cpp	2008-06-13 11:24:46.000000000 +0200
@@ -23,6 +23,7 @@
 #include "soundserver.h"
 #include "kartsserver.h"
 #include <stdio.h>
+#include <stdlib.h>
 #include <iostream>
 #include <fstream>
 #include <kdebug.h>
--- ./arts/modules/effects/synth_stereo_fir_equalizer_impl.cc.orig	2008-06-13 09:26:09.000000000 +0200
+++ ./arts/modules/effects/synth_stereo_fir_equalizer_impl.cc	2008-06-13 09:26:16.000000000 +0200
@@ -29,6 +29,7 @@
 #include "artsmoduleseffects.h"
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #include <kglobal.h>
 #include <klocale.h>
--- ./arts/modules/synth/synth_capture_wav_impl.cc.orig	2008-06-13 08:58:12.000000000 +0200
+++ ./arts/modules/synth/synth_capture_wav_impl.cc	2008-06-13 08:58:20.000000000 +0200
@@ -25,6 +25,8 @@
 #include "stdsynthmodule.h"
 #include "debug.h"
 #include <fcntl.h>
+#include <string.h>
+#include <stdlib.h>
 
 using namespace std;
 namespace Arts {
--- ./arts/modules/synth/synth_midi_test_impl.cc.orig	2008-06-13 08:58:53.000000000 +0200
+++ ./arts/modules/synth/synth_midi_test_impl.cc	2008-06-13 08:59:00.000000000 +0200
@@ -9,6 +9,7 @@
 #include "audiosubsys.h"
 #include <fstream>
 #include <math.h>
+#include <stdlib.h>
 
 using namespace Arts;
 using namespace std;
--- ./arts/modules/synth/synth_noise_impl.cc.orig	2008-06-13 08:55:08.000000000 +0200
+++ ./arts/modules/synth/synth_noise_impl.cc	2008-06-13 08:55:15.000000000 +0200
@@ -23,6 +23,8 @@
 #include "artsmodulessynth.h"
 #include "stdsynthmodule.h"
 
+#include <stdlib.h>
+
 using namespace Arts;
 
 namespace Arts {
--- ./arts/modules/synth/synth_osc_impl.cc.orig	2008-06-13 08:56:25.000000000 +0200
+++ ./arts/modules/synth/synth_osc_impl.cc	2008-06-13 08:56:31.000000000 +0200
@@ -27,6 +27,7 @@
 #include <gsl/gslsignal.h>
 
 #include <math.h>
+#include <string.h>
 
 using namespace Arts;
 
--- ./arts/modules/synth/synth_play_pat_impl.cc.orig	2008-06-13 08:57:29.000000000 +0200
+++ ./arts/modules/synth/synth_play_pat_impl.cc	2008-06-13 08:57:40.000000000 +0200
@@ -24,6 +24,8 @@
 #include <sys/stat.h>
 #include <stdsynthmodule.h>
 #include <unistd.h>
+#include <string.h>
+#include <stdlib.h>
 #include <math.h>
 #include <debug.h>
 #include <cache.h>
--- ./arts/modules/synth/synth_sequence_freq_impl.cc.orig	2008-06-13 09:04:38.000000000 +0200
+++ ./arts/modules/synth/synth_sequence_freq_impl.cc	2008-06-13 09:04:52.000000000 +0200
@@ -18,6 +18,7 @@
 */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include "artsmodulessynth.h"
 #include "stdsynthmodule.h"
 #include <debug.h>
--- ./arts/modules/synth/synth_sequence_impl.cc.orig	2008-06-13 09:00:54.000000000 +0200
+++ ./arts/modules/synth/synth_sequence_impl.cc	2008-06-13 09:01:03.000000000 +0200
@@ -26,6 +26,8 @@
     */
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include "artsmodulessynth.h"
 #include "stdsynthmodule.h"
 
--- ./arts/runtime/artsbuilderloader_impl.cc.orig	2008-06-13 08:25:32.000000000 +0200
+++ ./arts/runtime/artsbuilderloader_impl.cc	2008-06-13 08:25:39.000000000 +0200
@@ -26,6 +26,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <dirent.h>
 #include <set>
--- ./mpeglib/example/yaf/yafvorbis/vorbis_control.cpp.orig	2008-06-13 18:34:16.000000000 +0200
+++ ./mpeglib/example/yaf/yafvorbis/vorbis_control.cpp	2008-06-13 18:34:29.000000000 +0200
@@ -19,7 +19,7 @@
 
 
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
@@ -33,6 +33,7 @@
 #include <getopt.h>
 #endif
 
+using namespace std;
 
 
 
