diff -dur kdemultimedia-3.3.1/kioslave/audiocd/audiocd.cpp kdemultimedia-3.3.1-p/kioslave/audiocd/audiocd.cpp
--- kdemultimedia-3.3.1/kioslave/audiocd/audiocd.cpp	2004-10-03 07:42:35.000000000 +0000
+++ kdemultimedia-3.3.1-p/kioslave/audiocd/audiocd.cpp	2004-10-26 17:59:44.080754552 +0000
@@ -52,6 +52,19 @@
 #endif
 #undef __STRICT_ANSI__
 #include <asm/types.h>
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <linux/cdrom.h>
 #endif
 
diff -dur kdemultimedia-3.3.1/kscd/libwm/plat_linux.c kdemultimedia-3.3.1-p/kscd/libwm/plat_linux.c
--- kdemultimedia-3.3.1/kscd/libwm/plat_linux.c	2004-07-17 19:09:14.000000000 +0000
+++ kdemultimedia-3.3.1-p/kscd/libwm/plat_linux.c	2004-10-26 17:55:33.757809424 +0000
@@ -81,6 +81,19 @@
 #define asm __asm__
 #define inline __inline__
 #include <asm/types.h>
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <linux/cdrom.h>
 #undef asm
 #undef inline
diff -dur kdemultimedia-3.3.1/mpeglib/lib/input/cdromAccess_Linux.cpp kdemultimedia-3.3.1-p/mpeglib/lib/input/cdromAccess_Linux.cpp
--- kdemultimedia-3.3.1/mpeglib/lib/input/cdromAccess_Linux.cpp	2004-05-22 20:56:58.000000000 +0000
+++ kdemultimedia-3.3.1-p/mpeglib/lib/input/cdromAccess_Linux.cpp	2004-10-26 17:39:02.000000000 +0000
@@ -14,6 +14,19 @@
 #include "cdromToc.h"
 #include "cdromRawAccess.h"
 #include <asm/types.h>
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <linux/cdrom.h>
 #include <sys/ioctl.h>
 
