diff -Naur ./pcmciautils-013-orig/Makefile ./pcmciautils-013/Makefile
--- ./Makefile	2006-03-26 12:56:41.000000000 +0200
+++ ./Makefile	2006-03-26 20:47:52.240028024 +0200
@@ -73,7 +73,7 @@
 INSTALL_PROGRAM = ${INSTALL}
 INSTALL_DATA  = ${INSTALL} -m 644
 INSTALL_SCRIPT = ${INSTALL_PROGRAM}
-SYMLINK = /usr/bin/ln -sf
+SYMLINK = ln -sf
 
 # place to put our hotplug scripts nodes
 hotplugdir =	${etcdir}/hotplug
