diff -dur cups-1.2.4-orig/berkeley/Makefile cups-1.2.4/berkeley/Makefile
--- cups-1.2.4-orig/berkeley/Makefile	2006-03-05 17:48:12.000000000 +0100
+++ cups-1.2.4/berkeley/Makefile	2006-10-12 22:11:29.000000000 +0200
@@ -57,11 +57,11 @@
 
 install:	all
 	$(INSTALL_DIR) -m 755 $(BINDIR)
-	$(INSTALL_BIN) lpq $(BINDIR)
-	$(INSTALL_BIN) lpr $(BINDIR)
-	$(INSTALL_BIN) lprm $(BINDIR)
+	$(INSTALL_BIN) lpq $(BINDIR)/lpq_cups
+	$(INSTALL_BIN) lpr $(BINDIR)/lpr_cups
+	$(INSTALL_BIN) lprm $(BINDIR)/lprm_cups
 	$(INSTALL_DIR) -m 755 $(SBINDIR)
-	$(INSTALL_BIN) lpc $(SBINDIR)
+	$(INSTALL_BIN) lpc $(SBINDIR)/lpc_cups
 
 
 #
diff -dur cups-1.2.4-orig/man/Makefile cups-1.2.4/man/Makefile
--- cups-1.2.4-orig/man/Makefile	2006-07-12 22:45:13.000000000 +0200
+++ cups-1.2.4/man/Makefile	2006-10-12 22:11:29.000000000 +0200
@@ -107,27 +107,28 @@
 install:	all
 	$(INSTALL_DIR) -m 755 $(MANDIR)/man1
 	for file in $(MAN1); do \
-		echo Installing $$file in $(MANDIR)/man1...; \
-		$(INSTALL_MAN) $$file $(MANDIR)/man1; \
+		inst=`echo $$file | sed -e 's/\\.[0-9a-z]*$$//'`; \
+		$(INSTALL_MAN) $$file $(MANDIR)/man1/$$inst'_cups.1'; \
 	done
 	$(INSTALL_DIR) -m 755 $(MANDIR)/man5
 	for file in $(MAN5); do \
-		echo Installing $$file in $(MANDIR)/man5...; \
-		$(INSTALL_MAN) $$file $(MANDIR)/man5; \
+		inst=`echo $$file | sed -e 's/\\.[0-9a-z]*$$//'`; \
+		$(INSTALL_MAN) $$file $(MANDIR)/man5/$$inst'_cups.5'; \
 	done
 	$(INSTALL_DIR) -m 755 $(MANDIR)/man7
 	for file in $(MAN7); do \
-		echo Installing $$file in $(MANDIR)/man7...; \
-		$(INSTALL_MAN) $$file $(MANDIR)/man7; \
+		inst=`echo $$file | sed -e 's/\\.[0-9a-z]*$$//'`; \
+		$(INSTALL_MAN) $$file $(MANDIR)/man7/$$inst'_cups.7'; \
 	done
 	$(INSTALL_DIR) -m 755 $(AMANDIR)/man$(MAN8DIR)
 	for file in $(MAN8); do \
-		echo Installing $$file in $(AMANDIR)/man$(MAN8DIR)...; \
-		$(INSTALL_MAN) $$file $(AMANDIR)/man$(MAN8DIR); \
+		echo $$file; \
+		inst=`echo $$file | sed -e 's/\\.[0-9a-z]*$$//'`; \
+		$(INSTALL_MAN) $$file $(AMANDIR)/man8/$$inst'_cups'.$(MAN8EXT); \
 	done
-	$(RM) $(AMANDIR)/man$(MAN8DIR)/reject.$(MAN8EXT)
-	$(LN) accept.$(MAN8EXT) $(AMANDIR)/man$(MAN8DIR)/reject.$(MAN8EXT)
-	$(RM) $(AMANDIR)/man$(MAN8DIR)/cupsdisable.$(MAN8EXT)
+	$(RM) $(AMANDIR)/man$(MAN8DIR)/reject_cups.$(MAN8EXT)
+	$(LN) accept_cups.$(MAN8EXT) $(AMANDIR)/man$(MAN8DIR)/reject_cups.$(MAN8EXT)
+	$(RM) $(AMANDIR)/man8/cupsdisable.$(MAN8EXT)
 	$(LN) cupsenable.$(MAN8EXT) $(AMANDIR)/man$(MAN8DIR)/cupsdisable.$(MAN8EXT)
 
 
diff -dur cups-1.2.4-orig/systemv/Makefile cups-1.2.4/systemv/Makefile
--- cups-1.2.4-orig/systemv/Makefile	2006-06-05 17:25:23.000000000 +0200
+++ cups-1.2.4/systemv/Makefile	2006-10-12 22:14:25.000000000 +0200
@@ -60,25 +60,25 @@
 
 install:	all
 	$(INSTALL_DIR) -m 755 $(SBINDIR)
-	$(INSTALL_BIN) accept $(SBINDIR)
+	$(INSTALL_BIN) accept $(SBINDIR)/accept_cups
 	$(RM) $(SBINDIR)/reject
-	$(LN) accept $(SBINDIR)/reject
+	$(LN) accept $(SBINDIR)/reject_cups
 	$(INSTALL_BIN) cupsaddsmb $(SBINDIR)
-	$(INSTALL_BIN) lpadmin $(SBINDIR)
-	$(INSTALL_BIN) lpinfo $(SBINDIR)
-	$(INSTALL_BIN) lpmove $(SBINDIR)
+	$(INSTALL_BIN) lpadmin $(SBINDIR)/lpadmin_cups
+	$(INSTALL_BIN) lpinfo $(SBINDIR)/lpinfo_cups
+	$(INSTALL_BIN) lpmove $(SBINDIR)/lpmove_cups
 	$(INSTALL_DIR) -m 755 $(BINDIR)
-	$(INSTALL_BIN) cancel $(BINDIR)
+	$(INSTALL_BIN) cancel $(BINDIR)/cancel_cups
 	$(INSTALL_BIN) cupstestdsc $(BINDIR)
 	$(INSTALL_BIN) cupstestppd $(BINDIR)
 	$(RM) $(SBINDIR)/cupsdisable
 	$(LN) accept $(SBINDIR)/cupsdisable
 	$(RM) $(SBINDIR)/cupsenable
 	$(LN) accept $(SBINDIR)/cupsenable
-	$(INSTALL_BIN) lp $(BINDIR)
-	$(INSTALL_BIN) lpoptions $(BINDIR)
-	$(INSTALL_BIN) lpstat $(BINDIR)
-	$(INSTALL_BIN) -m 4755 -o root lppasswd $(BINDIR) || $(INSTALL_BIN) lppasswd $(BINDIR)
+	$(INSTALL_BIN) lp $(BINDIR)/lp_cups
+	$(INSTALL_BIN) lpoptions $(BINDIR)/lpoptions_cups
+	$(INSTALL_BIN) lpstat $(BINDIR)/lpstat_cups
+	$(INSTALL_BIN) -m 4755 -o root lppasswd $(BINDIR)/lppasswd_cups || $(INSTALL_BIN) lppasswd $(BINDIR)/lppasswd_cups
 
 
 #
