--- rss_glx-0.7.5/reallyslick/cpp_src/skyrocket_sound.cpp.orig	2005-07-11 17:22:55.000000000 +0200
+++ rss_glx-0.7.5/reallyslick/cpp_src/skyrocket_sound.cpp	2005-07-11 17:24:10.000000000 +0200
@@ -72,7 +72,7 @@
 
 	j = (unsigned char *)&i;
 
-	alutInit (NULL, 0);
+	alutInit (0, 0, 0);
 
 	alDistanceModel (AL_INVERSE_DISTANCE);
 	// As of October, 2001, AL_ROLLOFF_FACTOR isn't implemented in the Windows 
