--- ./configure.orig	2006-03-30 23:30:25.000000000 +0200
+++ ./configure	2006-03-30 23:30:40.000000000 +0200
@@ -290,7 +290,7 @@
     esac
 fi
 
-MP4_LDFLAGS="-lgpac_static"
+MP4_LDFLAGS="-lgpac"
 if [ $SYS = CYGWIN -o $SYS = MINGW ]; then
     MP4_LDFLAGS="$MP4_LDFLAGS -lwinmm"
 fi
--- ./muxers.c.orig	2006-03-30 23:39:03.000000000 +0200
+++ ./muxers.c	2006-03-30 23:39:15.000000000 +0200
@@ -647,7 +647,7 @@
         return -1;
 
     memset(p_mp4, 0, sizeof(mp4_t));
-    p_mp4->p_file = gf_isom_open(psz_filename, GF_ISOM_OPEN_WRITE, NULL);
+    p_mp4->p_file = gf_isom_open(psz_filename, GF_ISOM_OPEN_WRITE);
 
     if ((p_mp4->p_sample = gf_isom_sample_new()) == NULL)
     {
