--- Deva/examples/Makefile.orig	2004-08-30 00:02:12.000000000 +0200
+++ Deva/examples/Makefile	2004-08-30 00:08:40.000000000 +0200
@@ -12,46 +13,46 @@
 all:	${APS}
 
 ${DBIN}/DEVA_listTest:	DEVA_listTest.o
-	${CXX} ${CXXFLAG} DEVA_listTest.o -o ${DBIN}/DEVA_listTest ${LIBDIR} -ldeva_kernel
+	${CXX} ${CXXFLAG} DEVA_listTest.o -o ${DBIN}/DEVA_listTest ${LIBDIR} -ldeva_kernel ${LIBSTDCXX}
 
 ${DBIN}/DEVA_dictionaryTest:	DEVA_dictionaryTest.o
-	${CXX} ${CXXFLAG} DEVA_dictionaryTest.o -o ${DBIN}/DEVA_dictionaryTest ${LIBDIR} -ldeva_kernel
+	${CXX} ${CXXFLAG} DEVA_dictionaryTest.o -o ${DBIN}/DEVA_dictionaryTest ${LIBDIR} -ldeva_kernel ${LIBSTDCXX}
 
 ${DBIN}/DEVA_paramsTest:	DEVA_paramsTest.o
-	${CXX} ${CXXFLAG} DEVA_paramsTest.o -o ${DBIN}/DEVA_paramsTest ${LIBDIR} -ldeva_kernel -ldeva_comms
+	${CXX} ${CXXFLAG} DEVA_paramsTest.o -o ${DBIN}/DEVA_paramsTest ${LIBDIR} -ldeva_kernel -ldeva_comms ${LIBSTDCXX}
 
 
 ${DBIN}/DEVA_pathTest:	DEVA_pathTest.o
-	${CXX} ${CXXFLAG} DEVA_pathTest.o -o ${DBIN}/DEVA_pathTest ${LIBDIR} -ldeva_kernel
+	${CXX} ${CXXFLAG} DEVA_pathTest.o -o ${DBIN}/DEVA_pathTest ${LIBDIR} -ldeva_kernel ${LIBSTDCXX}
 
 ${DBIN}/DEVA_CRCtest:	DEVA_CRCtest.o
-	${CXX} ${CXXFLAG} DEVA_CRCtest.o -o ${DBIN}/DEVA_CRCtest ${LIBDIR} -ldeva_kernel
+	${CXX} ${CXXFLAG} DEVA_CRCtest.o -o ${DBIN}/DEVA_CRCtest ${LIBDIR} -ldeva_kernel ${LIBSTDCXX}
 
 ${DBIN}/DEVA_threadListTest:    DEVA_threadListTest.o
-	${CXX} ${CXXFLAG} DEVA_threadListTest.o -o ${DBIN}/DEVA_threadListTest ${LIBDIR} -ldeva_kernel -lpthread
+	${CXX} ${CXXFLAG} DEVA_threadListTest.o -o ${DBIN}/DEVA_threadListTest ${LIBDIR} -ldeva_kernel ${LIBSTDCXX} -lpthread
 
 
 ${DBIN}/DEVA_nameTest:	DEVA_nameTest.o
-	${CXX} ${CXXFLAG} DEVA_nameTest.o -o ${DBIN}/DEVA_nameTest ${LIBDIR} -ldeva_comms -ldeva_kernel
+	${CXX} ${CXXFLAG} DEVA_nameTest.o -o ${DBIN}/DEVA_nameTest ${LIBDIR} -ldeva_comms -ldeva_kernel ${LIBSTDCXX}
 
 ${DBIN}/DEVA_msgTest:	DEVA_msgTest.o
-	${CXX} ${CXXFLAG} DEVA_msgTest.o -o ${DBIN}/DEVA_msgTest ${LIBDIR} -ldeva_comms -ldeva_kernel
+	${CXX} ${CXXFLAG} DEVA_msgTest.o -o ${DBIN}/DEVA_msgTest ${LIBDIR} -ldeva_comms -ldeva_kernel ${LIBSTDCXX}
 
 
 ${DBIN}/DEVA_server:	DEVA_server.o
-	${CXX} ${CXXFLAG} DEVA_server.o -o ${DBIN}/DEVA_server ${LIBDIR} -ldeva_server -ldeva_comms -L${MAV_HOME}/lib -lmaverik -ldeva_kernel -lpthread ${DYNAMIC}
+	${CXX} ${CXXFLAG} DEVA_server.o -o ${DBIN}/DEVA_server ${LIBDIR} -ldeva_server -ldeva_comms -L${MAV_HOME}/lib -lmaverik -ldeva_kernel ${LIBSTDCXX} -lpthread ${DYNAMIC}
 
 ${DBIN}/DEVA_client:	DEVA_client.o
-	${CXX} ${CXXFLAGS} DEVA_client.o -o ${DBIN}/DEVA_client ${LIBDIR} -ldeva_viewer -L${MAV_HOME}/lib -lmaverik -ldeva_client -ldeva_comms -ldeva_kernel ${DYNAMIC}
+	${CXX} ${CXXFLAGS} DEVA_client.o -o ${DBIN}/DEVA_client ${LIBDIR} -ldeva_viewer -L${MAV_HOME}/lib -lmaverik -ldeva_client -ldeva_comms -ldeva_kernel ${LIBSTDCXX} ${DYNAMIC}
 
 ${DBIN}/DEVA_consoleBasic:	DEVA_consoleBasic.o
-	${CXX} ${CXXFLAG} DEVA_consoleBasic.o -o ${DBIN}/DEVA_consoleBasic ${LIBDIR} -ldeva_client -ldeva_comms -ldeva_kernel
+	${CXX} ${CXXFLAG} DEVA_consoleBasic.o -o ${DBIN}/DEVA_consoleBasic ${LIBDIR} -ldeva_client -ldeva_comms -ldeva_kernel ${LIBSTDCXX}
 
 ${DBIN}/DEVA_console:	DEVA_console.o
-	${CXX} ${CXXFLAG} DEVA_console.o -o ${DBIN}/DEVA_console ${LIBDIR} -ldeva_client -ldeva_comms -ldeva_kernel -lreadline -ltermcap
+	${CXX} ${CXXFLAG} DEVA_console.o -o ${DBIN}/DEVA_console ${LIBDIR} -ldeva_client -ldeva_comms -ldeva_kernel ${LIBSTDCXX} -lreadline -ltermcap
 
 ${DBIN}/DEVA_viewer:	DEVA_viewer.o
-	${CXX} ${CXXFLAG} DEVA_viewer.o -o ${DBIN}/DEVA_viewer ${LIBDIR} -ldeva_viewer -ldeva_client -ldeva_comms -ldeva_kernel
+	${CXX} ${CXXFLAG} DEVA_viewer.o -o ${DBIN}/DEVA_viewer ${LIBDIR} -ldeva_viewer -ldeva_client -ldeva_comms -ldeva_kernel ${LIBSTDCXX}
 
 
 
