--- ././minisip/gui/gtkgui/eggtrayicon.cxx.orig	2005-02-22 13:18:58.417257256 +0100
+++ ././minisip/gui/gtkgui/eggtrayicon.cxx	2005-02-22 13:19:15.680632824 +0100
@@ -20,7 +20,7 @@
 
 #ifndef WIN32
 #include <string.h>
-#include <gtk-2.0/gdk/gdkx.h>
+#include <gdk/gdkx.h>
 //#include<gtkmm.h>
 #include "eggtrayicon.h"
 
--- ././minisip/gui/gtkgui/eggtrayicon.h.orig	2005-02-22 13:19:06.194075000 +0100
+++ ././minisip/gui/gtkgui/eggtrayicon.h	2005-02-22 13:19:29.178580824 +0100
@@ -22,8 +22,8 @@
 #define __EGG_TRAY_ICON_H__
 
 #ifndef WIN32
-#include <gtk-2.0/gtk/gtkplug.h>
-#include <gtk-2.0/gdk/gdkx.h>
+#include <gtk/gtkplug.h>
+#include <gdk/gdkx.h>
 
 G_BEGIN_DECLS
 
