diff -ruN ksendfax-3.0.2/ksendfax/myscandialog.cpp ksendfax-3.0.2-new/ksendfax/myscandialog.cpp
--- ksendfax-3.0.2/ksendfax/myscandialog.cpp	2001-11-08 17:00:26.000000000 +0100
+++ ksendfax-3.0.2-new/ksendfax/myscandialog.cpp	2005-05-22 21:18:57.000000000 +0200
@@ -11,6 +11,7 @@
 #include <qstrlist.h>
 #include <qtooltip.h>
 #include <qsizepolicy.h> 
+#include <qcheckbox.h>
 
 #include <kglobal.h>
 #include <klocale.h>
