--- ./configure.orig	2004-09-07 15:06:21.021177544 +0200
+++ ./configure	2004-09-07 15:07:22.693801880 +0200
@@ -3272,8 +3272,7 @@
 
 pcap_dir=""
 for dir in $PCAP_DIR ; do
-    if test -d $dir -a -r "$dir/pcap.h" -a \
-                       -r "$dir/net/bpf.h" ; then
+    if test -d $dir -a -r "$dir/pcap.h" ; then
        if test -n "$pcap_dir" -a "$pcap_dir" != "$dir"; then
           echo
           echo; echo more than one set found in:
--- ./ngrep.c.orig	2004-09-07 15:08:07.421002312 +0200
+++ ./ngrep.c	2004-09-07 15:08:14.248964304 +0200
@@ -55,7 +55,6 @@
 #include <netinet/ip_icmp.h>
 
 #include <pcap.h>
-#include <net/bpf.h>
 
 #include <stdlib.h>
 #include <string.h>
