diff -dur prboom-2.3.1/src/m_swap.h prboom-2.3.1-p/src/m_swap.h
--- prboom-2.3.1/src/m_swap.h	2004-09-15 21:09:24.000000000 +0200
+++ prboom-2.3.1-p/src/m_swap.h	2004-11-06 10:37:14.347851896 +0100
@@ -49,6 +49,19 @@
 
 /* Try to use superfast macros on systems that support them */
 #ifdef HAVE_ASM_BYTEORDER_H
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <asm/byteorder.h>
 #ifdef __arch__swab16
 #define doom_swap_s  (signed short)__arch__swab16
