diff -dur k3b-0.11.17/src/device/k3bscsicommand.h k3b-0.11.17-p/src/device/k3bscsicommand.h
--- k3b-0.11.17/src/device/k3bscsicommand.h	2004-01-21 10:20:11.000000000 +0000
+++ k3b-0.11.17-p/src/device/k3bscsicommand.h	2004-10-26 20:54:50.798490672 +0000
@@ -18,6 +18,19 @@
 
 #include <sys/types.h>
 #undef __STRICT_ANSI__
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <linux/cdrom.h>
 #define __STRICT_ANSI__
 
