--- ./lib-src/libflac/include/FLAC/seekable_stream_decoder.h.orig	2006-04-28 10:00:26.000000000 +0200
+++ ./lib-src/libflac/include/FLAC/seekable_stream_decoder.h	2006-04-28 10:00:32.000000000 +0200
@@ -681,7 +681,7 @@
  * \retval FLAC__StreamDecoderState
  *    The stream decoder state.
  */
-FLAC__StreamDecoderState FLAC__seekable_stream_decoder_get_stream_decoder_state(const FLAC__SeekableStreamDecoder *decoder);
+FLAC__SeekableStreamDecoderState FLAC__seekable_stream_decoder_get_stream_decoder_state(const FLAC__SeekableStreamDecoder *decoder);
 
 /** Get the "MD5 signature checking" flag.
  *  This is the value of the setting, not whether or not the decoder is
--- ./lib-src/libflac/src/libFLAC++/seekable_stream_decoder.cc.orig	2006-04-28 10:02:47.000000000 +0200
+++ ./lib-src/libflac/src/libFLAC++/seekable_stream_decoder.cc	2006-04-28 10:03:39.000000000 +0200
@@ -99,7 +99,7 @@
 		Stream::State SeekableStream::get_stream_decoder_state() const
 		{
 			FLAC__ASSERT(is_valid());
-			return Stream::State(::FLAC__seekable_stream_decoder_get_stream_decoder_state(decoder_));
+			return Stream::State((FLAC__StreamDecoderState)::FLAC__seekable_stream_decoder_get_stream_decoder_state(decoder_));
 		}
 
 		bool SeekableStream::get_md5_checking() const
