--- ./Types.h.orig	Wed Sep  1 21:18:18 2004
+++ ./Types.h	Wed Sep  1 21:21:41 2004
@@ -74,11 +74,11 @@
 
   void clear()
   {
-    #if defined(__STL_USE_STD_ALLOCATORS) || defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(__STL_USE_STD_ALLOCATORS) || defined(_VECTOR_H)
 
-    #if defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(_VECTOR_H)
 
-    _Destroy(_M_start, _M_finish);
+    _Destroy(_M_impl._M_start, _M_impl._M_finish);
 
     #else
 
@@ -86,9 +86,9 @@
 
     #endif
 
-    _M_deallocate(_M_start, _M_end_of_storage - _M_start);
+    _M_deallocate(_M_impl._M_start, _M_impl._M_end_of_storage - _M_impl._M_start);
 
-    _M_start = _M_finish = _M_end_of_storage = 0;
+    _M_impl._M_start = _M_impl._M_finish = _M_impl._M_end_of_storage = 0;
 
     #else
 
@@ -108,21 +108,21 @@
 
   void clear()
   {
-    #if defined(__STL_USE_STD_ALLOCATORS) || defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(__STL_USE_STD_ALLOCATORS) || defined(_VECTOR_H)
 
-    #if defined(__GLIBCPP_INTERNAL_VECTOR_H)
+    #if defined(_VECTOR_H)
 
-    _Destroy(_M_start, _M_finish);
+    _Destroy(_M_impl._M_start, _M_impl._M_finish);
 
     #else
 
-    destroy(_M_start, _M_finish);
+    destroy(_M_impl._M_start, _M_impl._M_finish);
 
     #endif
 
-    _M_deallocate(_M_start, _M_end_of_storage - _M_start);
+    _M_deallocate(_M_impl._M_start, _M_impl._M_end_of_storage - _M_impl._M_start);
 
-    _M_start = _M_finish = _M_end_of_storage = 0;
+    _M_impl._M_start = _M_impl._M_finish = _M_impl._M_end_of_storage = 0;
 
     #else
 

--- ./Jpeg.cpp.orig	2004-09-13 03:41:49.905274168 +0200
+++ ./Jpeg.cpp	2004-09-13 03:43:06.886571232 +0200
@@ -452,7 +452,7 @@
     // next line.
     //
 
-    (char *) pixelPtr += RoundUp4(w * 2) - w * 2;
+    pixelPtr += RoundUp4(w * 2) - w * 2;
 
     dy++;
   }
@@ -562,7 +562,7 @@
       pixelPtr += 3;
     }
 
-    (char *) pixelPtr += RoundUp4(w * 3) - w * 3;
+    pixelPtr += RoundUp4(w * 3) - w * 3;
 
     dy++;
   }
--- ./Png.cpp.orig	2004-09-13 03:43:18.138860624 +0200
+++ ./Png.cpp	2004-09-13 03:43:41.532304280 +0200
@@ -452,7 +452,7 @@
     // next line
     //
 
-    (char *) pixelPtr += RoundUp4(w * 2) - w * 2;
+    pixelPtr += RoundUp4(w * 2) - w * 2;
   }
 
   png_destroy_read_struct(&png_ptr, &info_ptr,NULL);
@@ -578,7 +578,7 @@
       pixelPtr += 3;
     }
 
-    (char *) pixelPtr += RoundUp4(w * 3) - w * 3;
+    pixelPtr += RoundUp4(w * 3) - w * 3;
   }
 
   png_destroy_read_struct(&png_ptr, &info_ptr,NULL);
