--- util-linux-2.12h/partx/gpt.c	2002-08-04 00:20:16.000000000 +0000
+++ util-linux-2.12h-patch/partx/gpt.c	2004-10-23 05:51:20.169235312 +0000
@@ -33,6 +33,19 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <errno.h>
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <asm/byteorder.h>
 #include "crc32.h"
 #include "gpt.h"
