--- ./partx/addpart.c.orig	2004-12-30 18:27:39.994411784 +0100
+++ ./partx/addpart.c	2004-12-30 18:27:49.601951216 +0100
@@ -3,6 +3,7 @@
 #include <fcntl.h>
 #include <stdlib.h>
 #include <sys/ioctl.h>
+#include <linux/compiler.h>
 #include <linux/blkpg.h>
 
 int
--- ./partx/delpart.c.orig	2004-12-30 18:28:55.711900976 +0100
+++ ./partx/delpart.c	2004-12-30 18:29:03.194763408 +0100
@@ -3,6 +3,7 @@
 #include <fcntl.h>
 #include <stdlib.h>
 #include <sys/ioctl.h>
+#include <linux/compiler.h>
 #include <linux/blkpg.h>
 
 int
--- ./partx/partx.c.orig	2004-12-30 18:29:10.607636480 +0100
+++ ./partx/partx.c	2004-12-30 18:29:21.124037744 +0100
@@ -37,6 +37,7 @@
 #include <getopt.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
+#include <linux/compiler.h>
 #include <linux/hdreg.h>        /* HDIO_GETGEO */
 #include <linux/blkpg.h>
 #define BLKGETSIZE _IO(0x12,96)    /* return device size */
