--- ./libs/libmyth/mythcdrom-linux.cpp.orig	2004-10-25 16:22:08.000000000 +0200
+++ ./libs/libmyth/mythcdrom-linux.cpp	2004-10-25 17:32:04.000000000 +0200
@@ -1,3 +1,16 @@
+
+/* <do not do that at home kids> */
+/* simply setting _LOOSE_KERNEL_NAMES is not enough */
+#include <linux/types.h>
+
+extern "C" off_t lseek(int fildes, off_t offset, int whence);
+extern "C" ssize_t read(int fd, void *buf, size_t count);
+
+#define _UNISTD_H
+#define _SYS_TYPES_H
+#define _SYS_SELECT_H
+/* </do not do that at home kids> */
+
 #include "mythcdrom.h"
 #include <sys/ioctl.h>                // ioctls
 #include <linux/cdrom.h>        // old ioctls for cdrom
