--- ./lib/putenv.c.orig	2004-10-01 05:45:46.844122664 +0200
+++ ./lib/putenv.c	2004-10-01 05:45:56.619636560 +0200
@@ -37,8 +37,10 @@
    with a non-const argument.  That would conflict with the declaration of
    rpl_putenv below (due to the #define putenv rpl_putenv from config.h).  */
 
+#if 0
 void *malloc ();
 void free ();
+#endif
 
 #include <string.h>
 
--- ./lib/getline.c.orig	2004-10-01 05:46:07.911919872 +0200
+++ ./lib/getline.c	2004-10-01 05:46:56.900472480 +0200
@@ -34,11 +34,13 @@
 
 #if defined __GNU_LIBRARY__ && HAVE_GETDELIM
 
+#if 0
 ssize_t
 getline (char **lineptr, size_t *linesize, FILE *stream)
 {
   return getdelim (lineptr, linesize, '\n', stream);
 }
+#endif
 
 #else /* ! have getdelim */
 
