diff -dur e2fsprogs-1.38/e2fsck/Makefile.in e2fsprogs-1.38-p/e2fsck/Makefile.in
--- e2fsprogs-1.38/e2fsck/Makefile.in	2005-06-29 19:42:20.000000000 +0200
+++ e2fsprogs-1.38-p/e2fsck/Makefile.in	2005-10-18 20:48:27.000000000 +0200
@@ -16,7 +16,7 @@
 MANPAGES=	e2fsck.8
 XTRA_CFLAGS=	-DRESOURCE_TRACK
 
-LIBS= $(LIBEXT2FS) $(LIBCOM_ERR) $(LIBBLKID) $(LIBUUID) $(LIBINTL)
+LIBS= $(LIBE2P) $(LIBEXT2FS) $(LIBCOM_ERR) $(LIBBLKID) $(LIBUUID) $(LIBINTL)
 DEPLIBS= $(LIBEXT2FS) $(LIBCOM_ERR) $(LIBBLKID) $(DEPLIBUUID)
 
 STATIC_LIBS= $(STATIC_LIBEXT2FS) $(STATIC_LIBCOM_ERR) $(STATIC_LIBBLKID) \
diff -dur e2fsprogs-1.38/tests/progs/Makefile.in e2fsprogs-1.38-p/tests/progs/Makefile.in
--- e2fsprogs-1.38/tests/progs/Makefile.in	2005-06-29 19:42:20.000000000 +0200
+++ e2fsprogs-1.38-p/tests/progs/Makefile.in	2005-10-18 20:48:27.000000000 +0200
@@ -21,7 +21,7 @@
 
 SRCS=	$(srcdir)/test_rel.c 
 
-LIBS= $(LIBEXT2FS) $(LIBSS) $(LIBCOM_ERR)
+LIBS= $(LIBE2P) $(LIBEXT2FS) $(LIBSS) $(LIBCOM_ERR)
 DEPLIBS= $(LIBEXT2FS) $(LIBSS) $(LIBCOM_ERR)
 
 .c.o:
