diff -dur gst-plugins-0.8.2/sys/cdrom/gstcdplayer_ioctl.c gst-plugins-0.8.2-p/sys/cdrom/gstcdplayer_ioctl.c
--- gst-plugins-0.8.2/sys/cdrom/gstcdplayer_ioctl.c	2004-03-14 22:34:30.000000000 +0000
+++ gst-plugins-0.8.2-p/sys/cdrom/gstcdplayer_ioctl.c	2004-10-25 18:30:52.973436168 +0000
@@ -40,6 +40,19 @@
 static gint cddb_sum (gint n);
 
 #if defined(HAVE_LINUX_CDROM_H)
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <linux/cdrom.h>
 #elif defined(HAVE_SYS_CDIO_H)
 #include <sys/cdio.h>
diff -dur gst-plugins-0.8.2/sys/vcd/vcdsrc.c gst-plugins-0.8.2-p/sys/vcd/vcdsrc.c
--- gst-plugins-0.8.2/sys/vcd/vcdsrc.c	2004-05-30 12:26:23.000000000 +0000
+++ gst-plugins-0.8.2-p/sys/vcd/vcdsrc.c	2004-10-25 18:30:11.102801472 +0000
@@ -32,6 +32,19 @@
 #include <sys/types.h>
 #include <dirent.h>
 #include <errno.h>
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <linux/cdrom.h>
 
 #include <vcdsrc.h>
