diff -ruN kdeadmin-3.5.0/configure kdeadmin-3.5.0-new/configure
--- kdeadmin-3.5.0/configure	2005-11-20 20:59:57.000000000 +0100
+++ kdeadmin-3.5.0-new/configure	2005-12-07 22:05:23.000000000 +0100
@@ -34197,7 +34197,7 @@
 #define HAVE_PAM 1
 _ACEOF
 
-      PAMLIBS="$PAM_MISC_LIB -lpam $LIBDL"
+      PAMLIBS="$PAM_MISC_LIB -lpam -lpam_misc $LIBDL"
       use_pam=yes
 
             if test "$pam_header" = "pam/pam_appl.h"; then
diff -ruN kdeadmin-3.5.0/kuser/Makefile.in kdeadmin-3.5.0-new/kuser/Makefile.in
--- kdeadmin-3.5.0/kuser/Makefile.in	2005-11-20 20:59:55.000000000 +0100
+++ kdeadmin-3.5.0-new/kuser/Makefile.in	2005-12-07 22:06:08.000000000 +0100
@@ -409,7 +409,7 @@
 # May be -D_XU_QUOTA -D_XU_SHADOW
 #AM_CXXFLAGS = 	$(KU_NIS) $(KU_QUOTA) $(QUOTACTL) -D_KU_DEBUG
 kuser_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kuser_LDADD = $(LIB_KDEUI) $(LIBCRYPT) $(LIBSHADOW) $(LIBGEN) $(LIB_KABC) -lkntlm
+kuser_LDADD = $(LIB_KDEUI) $(LIBCRYPT) $(LIBSHADOW) $(PAMLIBS) $(LIBGEN) $(LIB_KABC) -lkntlm
 
 ####### Files
 #>- kuser_SOURCES = \
