diff -dur cdrdao-1.1.9/scsilib/libscg/scsi-linux-sg.c cdrdao-1.1.9-p/scsilib/libscg/scsi-linux-sg.c
--- cdrdao-1.1.9/scsilib/libscg/scsi-linux-sg.c	2004-06-07 01:04:12.000000000 +0000
+++ cdrdao-1.1.9-p/scsilib/libscg/scsi-linux-sg.c	2004-10-25 17:53:24.944188560 +0000
@@ -69,6 +69,19 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 0)
 	#define __KERNEL__
 	#include <asm/types.h>
+
+	#ifdef __CHECKER__
+	#define __bitwise __attribute__((bitwise))
+	#else
+	#define __bitwise
+	#endif
+	#define __le16 __bitwise __u16
+	#define __be16 __bitwise __u16
+	#define __le32 __bitwise __u32
+	#define __be32 __bitwise __u32
+	#define __le64 __bitwise __u64
+	#define __be64 __bitwise __u64
+	
 	#include <asm/byteorder.h>
 	#undef __KERNEL__
 #endif
Nur in cdrdao-1.1.9-p/scsilib/libscg: scsi-linux-sg.c~.
