--- ./Makefile.in.orig	2004-09-19 16:21:17.000000000 +0200
+++ ./Makefile.in	2005-02-06 01:01:54.046888768 +0100
@@ -226,8 +226,8 @@
 
 @HAVE_DOXYGEN_FALSE@DOC_DIR = 
 @HAVE_DOXYGEN_TRUE@DOC_DIR = doc
-SUBDIRS = config jack libjack jackd drivers example-clients $(DOC_DIR)
-DIST_SUBDIRS = config jack libjack jackd drivers example-clients doc
+SUBDIRS = config jack libjack jackd drivers $(DOC_DIR)
+DIST_SUBDIRS = config jack libjack jackd drivers doc
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = jack.pc
 EXTRA_DIST = COPYING COPYING.GPL COPYING.LGPL
