diff -Naur xemacs-21.4.15.orig/lib-src/movemail.c xemacs-21.4.15/lib-src/movemail.c
--- xemacs-21.4.15.orig/lib-src/movemail.c	2002-08-20 12:34:22.000000000 +0100
+++ xemacs-21.4.15/lib-src/movemail.c	2005-02-09 09:34:12.000000000 +0000
@@ -796,7 +796,7 @@
           mbx_delimit_begin (mbf);
 	  if (pop_retr (server, i, mbx_write, mbf) != POP_RETRIEVED)
 	    {
-	      error (Errmsg, NULL, NULL);
+	      error ("%s", Errmsg, NULL, NULL);
 	      close (mbfi);
 	      return (1);
 	    }
