diff -pruN etcnet-0.8.5/contrib/50-RockLinux-3 etcnet-0.8.5-new/contrib/50-RockLinux-3
--- etcnet-0.8.5/contrib/50-RockLinux-3	2006-09-26 20:17:55.000000000 +0200
+++ etcnet-0.8.5-new/contrib/50-RockLinux-3	2006-10-28 12:41:42.000000000 +0200
@@ -8,4 +8,4 @@ DHCP_HOSTNAME=
 DHCP_TIMEOUT=
 
 # Temp hack
-IFRENAME=/usr/sbin/ifrename
+IFRENAME=/sbin/ifrename
diff -pruN etcnet-0.8.5/etc/net/scripts/network.init etcnet-0.8.5-new/etc/net/scripts/network.init
--- etcnet-0.8.5/etc/net/scripts/network.init	2006-09-08 21:29:00.000000000 +0200
+++ etcnet-0.8.5-new/etc/net/scripts/network.init	2006-10-28 12:43:14.000000000 +0200
@@ -1,8 +1,8 @@
 #!/bin/bash
 #
-# network
-# chkconfig: 2345 10 90
-# description: /etc/net is network configuration software
+# Desc: Network configuration using etc/net
+# Runlevel: 15 rcX rc3 rc4 rc5
+#
 
 # ALTLinux: Do not load RH compatibility interface.
 WITHOUT_RC_COMPAT=1
