--- lshw-A.01.07/cdrom.cc	2004-05-11 22:14:35.000000000 +0000
+++ lshw-A.01.07-p/cdrom.cc	2004-10-28 13:11:56.366827712 +0000
@@ -4,6 +4,19 @@
 #include <sys/ioctl.h>
 #include <fcntl.h>
 #include <unistd.h>
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include <linux/cdrom.h>
 
 static char *id = "@(#) $Id: cdrom.cc 598 2004-05-11 22:14:35Z ezix $";
