--- ./src/scenes.c.orig       2004-05-09 18:38:40.000000000 +0200
+++ ./src/scenes.c     2004-08-10 09:31:45.750402978 +0200
@@ -60,7 +60,6 @@
                          *LED_yellow = NULL, *LED_red = NULL;
 
 
-void set_EQ_curve_values ();
 
 
 /*  Initialize all scene related structures and get the widget addresses.  */
@@ -210,7 +209,7 @@
 
                         s_crossfade_to_state (&scene_state[i], -1.0f);
 
-                        set_EQ_curve_values ();
+                        set_EQ_curve_values (0, 0.0);
 
                         s_history_add_state (scene_state[i]);
