--- ./build/CONFIG.orig	2004-11-24 15:37:27.887547136 +0100
+++ ./build/CONFIG	2004-11-24 15:38:25.342812608 +0100
@@ -2,8 +2,8 @@
 # (C) 2000 Sebastien GODARD (sysstat <at> wanadoo.fr)
 
 # Directories
-PREFIX = /usr/local
-SA_LIB_DIR = /usr/local/lib/sa
+PREFIX = /usr
+SA_LIB_DIR = /usr/lib/sa
 SA_DIR = /var/log/sa
 MAN_DIR = ${PREFIX}/man
 CLEAN_SA_DIR = n
--- ./build/CONFIG.def.orig	2004-11-24 15:38:02.224327152 +0100
+++ ./build/CONFIG.def	2004-11-24 15:38:19.258737528 +0100
@@ -2,8 +2,8 @@
 # (C) 2000 Sebastien GODARD (sysstat <at> wanadoo.fr)
 
 # Directories
-PREFIX = /usr/local
-SA_LIB_DIR = /usr/local/lib/sa
+PREFIX = /usr
+SA_LIB_DIR = /usr/lib/sa
 SA_DIR = /var/log/sa
 MAN_DIR = ${PREFIX}/man
 CLEAN_SA_DIR = n
--- ./build/Configure.sh.orig	2004-11-24 15:39:45.218669632 +0100
+++ ./build/Configure.sh	2004-11-24 15:39:01.481318720 +0100
@@ -10,10 +10,10 @@
 
 # Installation directory
 
-PREFIX=`${ASK} 'Installation directory:' "/usr/local" "prefix"`
+PREFIX=`${ASK} 'Installation directory:' "/usr" "prefix"`
 if [ ! -d ${PREFIX} ]; then
-	echo "WARNING: Directory ${PREFIX} not found: Using default (/usr/local)."
-	PREFIX=/usr/local
+	echo "WARNING: Directory ${PREFIX} not found: Using default (/usr)."
+	PREFIX=/usr
 fi
 
 # sadc directory
