--- ./src/qjackctlConnectionsForm.ui.orig	2005-10-28 13:33:05.000000000 +0200
+++ ./src/qjackctlConnectionsForm.ui	2005-10-28 13:33:11.000000000 +0200
@@ -439,7 +439,5 @@
 <pixmapinproject/>
 <layoutdefaults spacing="4" margin="4"/>
 <includehints>
-    <includehint>qjackctlconnectview.h</includehint>
-    <includehint>qjackctlconnectview.h</includehint>
 </includehints>
 </UI>
--- ./src/qjackctlPatchbayForm.ui.orig	2005-10-28 13:37:42.000000000 +0200
+++ ./src/qjackctlPatchbayForm.ui	2005-10-28 13:37:50.000000000 +0200
@@ -885,6 +885,5 @@
 <pixmapinproject/>
 <layoutdefaults spacing="4" margin="4"/>
 <includehints>
-    <includehint>qjackctlpatchbayview.h</includehint>
 </includehints>
 </UI>
