--- ./include/asm-ppc/bitops.h.orig	2004-09-02 06:46:36.005575240 +0200
+++ ./include/asm-ppc/bitops.h	2004-09-02 06:49:30.138103072 +0200
@@ -2,11 +2,11 @@
  * bitops.h: Bit string operations on the ppc
  */
 
-#ifdef __KERNEL__
 #ifndef _PPC_BITOPS_H
 #define _PPC_BITOPS_H
 
 #include <linux/config.h>
+#include <asm/ppc_asm.h>
 #include <asm/byteorder.h>
 #include <asm/atomic.h>
 
@@ -370,4 +370,3 @@
 #define minix_find_first_zero_bit(addr,size) ext2_find_first_zero_bit(addr,size)
 
 #endif /* _PPC_BITOPS_H */
-#endif /* __KERNEL__ */
