diff -dur librep-0.17/configure librep-0.17-p/configure
--- librep-0.17/configure	2003-11-25 04:46:51.000000000 +0100
+++ librep-0.17-p/configure	2004-11-09 16:24:40.124499080 +0100
@@ -19377,11 +19377,12 @@
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char mpz_init ();
+#include <gmp.h>
 int
 main ()
 {
-mpz_init ();
+  mpz_t integ;
+  mpz_init(integ)
   ;
   return 0;
 }
Nur in librep-0.17-p/: configure~.
Nur in librep-0.17-p/: configure.rej.
