--- ./wx-config.in.orig	2004-11-22 02:40:51.785674568 +0100
+++ ./wx-config.in	2004-11-22 02:41:16.329943272 +0100
@@ -959,7 +959,7 @@
     if [ "x$this_linkage" = "x-static" ]; then
         wx_libs="$wx_libs ${libdir}/libwx_@TOOLCHAIN_NAME@.a $ldlibs_core @LIBS@"
     else
-        wx_libs="$wx_libs -lwx_@TOOLCHAIN_NAME@"
+        wx_libs="$wx_libs -lwx_@TOOLCHAIN_NAME@ $ldlibs_core @LIBS@"
     fi
 
     using_gui=yes
